--- a/axfer/container.h.org
+++ b/axfer/container.h
@@ -15,6 +15,8 @@
 #include <stdbool.h>
 #include <stdint.h>
 
+#include <assert.h>
+
 #include <alsa/asoundlib.h>
 
 enum container_type {
--- a/configure.ac.org
+++ b/configure.ac
@@ -3,12 +3,12 @@
 AC_INIT(alsa-utils, 1.2.6)
 AC_CONFIG_SRCDIR([aplay/aplay.c])
 AC_PREFIX_DEFAULT(/usr)
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([subdir-objects])
 
 AM_MAINTAINER_MODE([enable])
 
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.19.8])
+AM_GNU_GETTEXT_VERSION([0.19.7])
 
 dnl Checks for programs.
 

