--- enigma2-nightly.org/configure.ac
+++ enigma2-nightly/configure.ac
@@ -372,6 +372,7 @@
 include/Makefile
 lib/Makefile
 lib/base/eenv.cpp
+lib/libeplayer3/Makefile
 lib/python/Makefile
 lib/python/Components/Makefile
 lib/python/Components/Converter/Makefile
--- enigma2-tango.git/lib/libeplayer3/input.cpp
+++ enigma2-nightly/lib/libeplayer3/input.cpp
@@ -442,7 +442,8 @@
 	if (player->noprobe || player->isHttp) {
 #if (LIBAVFORMAT_VERSION_MAJOR <  55) || \
     (LIBAVFORMAT_VERSION_MAJOR == 55 && LIBAVFORMAT_VERSION_MINOR <  43) || \
-    (LIBAVFORMAT_VERSION_MAJOR == 55 && LIBAVFORMAT_VERSION_MINOR == 43 && LIBAVFORMAT_VERSION_MICRO < 100)
+    (LIBAVFORMAT_VERSION_MAJOR == 55 && LIBAVFORMAT_VERSION_MINOR == 43 && LIBAVFORMAT_VERSION_MICRO < 100) || \
+    (LIBAVFORMAT_VERSION_MAJOR == 57 && LIBAVFORMAT_VERSION_MINOR == 25)
 		avfc->max_analyze_duration = 1;
 #else
 		avfc->max_analyze_duration2 = 1;
--- enigma2-nightly.org/lib/python/Screens/InfoBar.py
+++ enigma2-nightly/lib/python/Screens/InfoBar.py
@@ -186,19 +186,21 @@
 
 	def toogleTvRadio(self): 
 		service = self.session.nav.getCurrentService()
-		info = service.info()
-		AudioPID = info.getInfo(iServiceInformation.sAudioPID)
-		VideoPID = info.getInfo(iServiceInformation.sVideoPID)
-
-		print "sAudioPID", AudioPID
-		print "sVideoPID", VideoPID
-
-		if VideoPID == -1:
-			print "Radio->TV"
-			self.showTv2()
-		else:
-			print "TV->Radio"
-			self.showRadio2()
+		if service:
+			info = service.info()
+			if info:
+				AudioPID = info.getInfo(iServiceInformation.sAudioPID)
+				VideoPID = info.getInfo(iServiceInformation.sVideoPID)
+
+				print "sAudioPID", AudioPID
+				print "sVideoPID", VideoPID
+
+				if VideoPID == -1:
+					print "Radio->TV"
+					self.showTv2()
+				else:
+					print "TV->Radio"
+					self.showRadio2()
 
 	def serviceStarted(self):  #override from InfoBarShowHide
 		new = self.servicelist.newServicePlayed()
