--- a/uapi/linux/netlink.h_org
+++ b/uapi/linux/netlink.h
@@ -35,7 +35,7 @@
 #define MAX_LINKS 32		
 
 struct sockaddr_nl {
-	__kernel_sa_family_t	nl_family;	/* AF_NETLINK	*/
+	sa_family_t	nl_family;	/* AF_NETLINK	*/
 	unsigned short	nl_pad;		/* zero		*/
 	__u32		nl_pid;		/* port ID	*/
        	__u32		nl_groups;	/* multicast groups mask */
--- a/uapi/linux/rtnetlink.h_org
+++ b/uapi/linux/rtnetlink.h
@@ -430,7 +430,7 @@
 
 /* RTA_VIA */
 struct rtvia {
-	__kernel_sa_family_t	rtvia_family;
+	sa_family_t		rtvia_family;
 	__u8			rtvia_addr[0];
 };
 
