--- a/common/glib-gen.mak	2019-04-19 11:16:18.000000000 +0200
+++ b/common/glib-gen.mak.patched	2024-03-02 20:47:02.774711988 +0100
@@ -5,7 +5,8 @@
 #glib_enum_define=GST_COLOR_BALANCE
 #glib_enum_prefix=gst_color_balance
 
-enum_headers=$(foreach h,$(glib_enum_headers),\n\#include \"$(h)\")
+hash:=\#
+enum_headers=$(foreach h,$(glib_enum_headers),\n$(hash)include \"$(h)\")
 
 # these are all the rules generating the relevant files
 %-marshal.h: %-marshal.list--- a/common/gst-glib-gen.mak	2024-03-02 21:16:44.173251077 +0100
--- a/common/gst-glib-gen.mak	2019-04-19 11:16:18.000000000 +0200
+++ b/common/gst-glib-gen.mak.patched	2024-03-02 20:47:04.774711988 +0100
@@ -8,7 +8,8 @@
 #glib_gen_decl_banner=GST_EXPORT
 #glib_gen_decl_include=\#include <gst/foo/foo-prelude.h>
 
-enum_headers=$(foreach h,$(glib_enum_headers),\n\#include \"$(h)\")
+hash:=\#
+enum_headers=$(foreach h,$(glib_enum_headers),\n$(hash)include \"$(h)\")
 
 # these are all the rules generating the relevant files
 $(glib_gen_basename)-marshal.h: $(glib_gen_basename)-marshal.list

