--- a/fribidi.pc.in
+++ b/fribidi.pc.in
@@ -1,9 +1,9 @@
 # @configure_input@
 
 prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
 
 Name: @PACKAGE_NAME@
 Description: Unicode Bidirectional Algorithm Library
--- a/lib/fribidi-types.h
+++ b/lib/fribidi-types.h
@@ -27,9 +27,11 @@
 #ifndef _FRIBIDI_TYPES_H
 #define _FRIBIDI_TYPES_H
 
+/*
 #ifdef HAVE_CONFIG_H
 # include <config.h>
 #endif
+*/
 
 #include "fribidi-common.h"
 

