--- a/configure.ac~
+++ b/configure.ac
@@ -32,12 +32,12 @@
 AM_MAINTAINER_MODE
 AC_ARG_VAR([ACLOCAL_FLAGS], [aclocal flags, e.g. -I <macro dir>])
 
-MM_PREREQ([0.9.10])
+MM_PREREQ([0.9.9])
 MM_INIT_MODULE([sigc++-2.0])
 MM_CONFIG_DOCTOOL_DIR([docs])
 
 AC_PROG_CXX
-MM_AX_CXX_COMPILE_STDCXX([11], [noext], [mandatory])
+MM_AX_CXX_COMPILE_STDCXX_11([noext], [mandatory])
 
 AC_DISABLE_STATIC
 LT_INIT([win32-dll])
 

