--- a/libxml.h
+++ b/libxml.h
@@ -13,10 +13,13 @@
 
 #ifndef NO_LARGEFILE_SOURCE
 #ifndef _LARGEFILE_SOURCE
 #define _LARGEFILE_SOURCE
 #endif
+#ifndef _LARGEFILE64_SOURCE
+#define _LARGEFILE64_SOURCE
+#endif
 #ifndef _FILE_OFFSET_BITS
 #define _FILE_OFFSET_BITS 64
 #endif
 #endif
 
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,9 +2,9 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = include . doc example xstc $(PYTHON_SUBDIR)
-
-DIST_SUBDIRS = include . doc example python xstc
+SUBDIRS = include . $(PYTHON_SUBDIR)
+
+DIST_SUBDIRS = include . python
 
 AM_CPPFLAGS = -I$(top_builddir)/include -I$(srcdir)/include
 
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -25,7 +25,7 @@
 
 libxml2mod_la_SOURCES = libxml.c libxml_wrap.h libxml2-py.h libxml2-py.c types.c
 libxml2mod_la_LDFLAGS = $(CYGWIN_EXTRA_LDFLAGS) $(WIN32_EXTRA_LDFLAGS) -module -avoid-version \
-        $(top_builddir)/libxml2.la $(CYGWIN_EXTRA_PYTHON_LIBADD) $(WIN32_EXTRA_PYTHON_LIBADD) $(PYTHON_LIBS)
+        $(top_builddir)/libxml2.la $(CYGWIN_EXTRA_PYTHON_LIBADD) $(WIN32_EXTRA_PYTHON_LIBADD) ../../../tufsbox/cdkroot/usr/lib/python2.7
 
 BUILT_SOURCES = libxml2-export.c libxml2-py.h libxml2-py.c

