--- e2openplugin-OpenWebif/plugin/controllers/models/info.py.org
+++ e2openplugin-OpenWebif/plugin/controllers/models/info.py
@@ -347,7 +347,10 @@
 		friendlychipsettext = "Broadcom " + friendlychipsettext
 	if not (info['fp_version'] is None or info['fp_version'] == 0):
 		friendlychipsetdescription = friendlychipsetdescription + " (" + _("Front processor version") + ")"
-		friendlychipsettext = friendlychipsettext + " (" + str(info['fp_version']) + ")"
+		if info['fp_version'].find(".") == -1:
+			friendlychipsettext = friendlychipsettext + " (" + str(info['fp_version']) + ")"
+		else:
+			friendlychipsettext = friendlychipsettext + " (" + info['fp_version'] + ")"
 
 	info['friendlychipsetdescription'] = friendlychipsetdescription
 	info['friendlychipsettext'] = friendlychipsettext
--- e2openplugin-OpenWebif/plugin/controllers/models/owibranding.py.org
+++ e2openplugin-OpenWebif/plugin/controllers/models/owibranding.py
@@ -145,69 +145,19 @@
 	if tpmloaded:
 		orgdream = tpm_check()
 
-	if fileExists("/proc/stb/info/hwmodel"):
-		brand = "DAGS"
-		f = open("/proc/stb/info/hwmodel", 'r')
-		procmodel = f.readline().strip()
+	if fileExists("/proc/stb/info/brand"):
+		f = open("/proc/stb/info/brand", 'r')
+		brand = f.readline().strip()
 		f.close()
-		if (procmodel.startswith("optimuss") or procmodel.startswith("pingulux")):
-			brand = "Edision"
-			model = procmodel.replace("optimmuss", "Optimuss ").replace("plus", " Plus").replace(" os", " OS")
-		elif (procmodel.startswith("fusion") or procmodel.startswith("purehd") or procmodel.startswith("revo4k") or procmodel.startswith("galaxy4k")):
-			brand = "Xsarius"
-			if procmodel == "fusionhd":
-				model = procmodel.replace("fusionhd", "Fusion HD")
-			elif procmodel == "fusionhdse":
-				model = procmodel.replace("fusionhdse", "Fusion HD SE")
-			elif procmodel == "purehd":
-				model = procmodel.replace("purehd", "Pure HD")
-			elif procmodel == "purehdse":
-					model = procmodel.replace("purehdse", "Pure HD SE")
-			elif procmodel == "revo4k":
-				model = procmodel.replace("revo4k", "Revo4K")
-			elif procmodel == "galaxy4k":
-				model = procmodel.replace("galaxy4k", "Galaxy4K")
-		elif (procmodel.startswith("lunix") or procmodel.startswith("dual")):
-			brand = "Qviart"
-			if procmodel == "lunix3-4k":
-				model = procmodel.replace("lunix3-4k", "Lunix3-4K")
-			elif procmodel == "lunix":
-				model = procmodel.replace("lunix", "Lunix")
-			elif procmodel == "lunix4k":
-				model = procmodel.replace("lunix4k", "Lunix4K")
-			elif procmodel == "dual":
-				model = procmodel.replace("dual", "Dual")
-	elif fileExists("/proc/stb/info/azmodel"):
-		brand = "AZBox"
-		f = open("/proc/stb/info/model", 'r')  # To-Do: Check if "model" is really correct ...
-		procmodel = f.readline().strip()
-		f.close()
-		model = procmodel.lower()
-	elif fileExists("/proc/stb/info/gbmodel"):
-		brand = "GigaBlue"
-		f = open("/proc/stb/info/gbmodel", 'r')
-		procmodel = f.readline().strip()
-		f.close()
-		if procmodel == "GBQUAD PLUS":
-			model = procmodel.replace("GBQUAD", "Quad").replace("PLUS", " Plus")
-		elif procmodel == "gbquad4k":
-			model = procmodel.replace("gbquad4k", "UHD Quad 4k")
-		elif procmodel == "quad4k":
-			model = procmodel.replace("quad4k", "UHD Quad 4k")
-		elif procmodel == "gbue4k":
-			model = procmodel.replace("gbue4k", "UHD UE 4k")
-		elif procmodel == "ue4k":
-			model = procmodel.replace("ue4k", "UHD UE 4k")
-		elif procmodel == "gbtrio4k":
-			model = procmodel.replace("gbtrio4k", "UHD Trio 4k")
-	elif fileExists("/proc/stb/info/vumodel") and not fileExists("/proc/stb/info/boxtype"):
-		brand = "Vu+"
-		f = open("/proc/stb/info/vumodel", 'r')
-		procmodel = f.readline().strip()
-		f.close()
-		model = procmodel.title().replace("olose", "olo SE").replace("olo2se", "olo2 SE").replace("2", "²").replace("4Kse", "4K SE")
-		if not procmodel.startswith("vu"):
-			procmodel = "vu%s" % procmodel
+		model = "unknown"
+		if fileExists("/proc/stb/info/model_name"):
+			f = open("/proc/stb/info/model_name", 'r')
+			model = f.readline().strip()
+			f.close()
+		if fileExists("/proc/stb/info/model"):
+			f = open("/proc/stb/info/model", 'r')
+			procmodel = f.readline().strip()
+			f.close()
 	elif fileExists("/proc/boxtype"):
 		f = open("/proc/boxtype", 'r')
 		procmodel = f.readline().strip().lower()
@@ -232,123 +182,7 @@
 		f = open("/proc/stb/info/boxtype", 'r')
 		procmodel = f.readline().strip().lower()
 		f.close()
-		if procmodel.startswith("et"):
-			if procmodel == "et7000mini":
-				brand = "Galaxy Innovations"
-				model = "ET-7000 Mini"
-			elif procmodel == "et11000":
-				brand = "Galaxy Innovations"
-				model = "ET-11000"
-			else:
-				brand = "Xtrend"
-				model = procmodel.upper()
-		elif procmodel.startswith("xpeed"):
-			brand = "Golden Interstar"
-			model = procmodel
-		elif procmodel.startswith("xp"):
-			brand = "MaxDigital"
-			model = procmodel.upper()
-		elif procmodel.startswith("ixuss"):
-			brand = "Medialink"
-			model = procmodel.replace(" ", "")
-		elif procmodel == "formuler4turbo":
-			brand = "Formuler"
-			model = "F4 Turbo"
-		elif procmodel.startswith("formuler"):
-			brand = "Formuler"
-			model = procmodel.replace("formuler", "")
-			if model.isdigit():
-				model = 'F' + model
-		elif procmodel.startswith("mbtwinplus"):
-			brand = "Miraclebox"
-			model = "Premium Twin+"
-		elif procmodel.startswith("alphatriplehd"):
-			brand = "SAB"
-			model = "Alpha Triple HD"
-		elif procmodel in ("7000s", "mbmicro"):
-			procmodel = "mbmicro"
-			brand = "Miraclebox"
-			model = "Premium Micro"
-		elif procmodel in ("7005s", "mbmicrov2"):
-			procmodel = "mbmicrov2"
-			brand = "Miraclebox"
-			model = "Premium Micro v2"
-		elif procmodel.startswith("ini"):
-			if procmodel.endswith("9000ru"):
-				brand = "Sezam"
-				model = "Marvel"
-			elif procmodel.endswith("5000ru"):
-				brand = "Sezam"
-				model = "hdx"
-			elif procmodel.endswith("1000ru"):
-				brand = "Sezam"
-				model = "hde"
-			elif procmodel.endswith("5000sv"):
-				brand = "Miraclebox"
-				model = "mbtwin"
-			elif procmodel.endswith("1000sv"):
-				brand = "Miraclebox"
-				model = "mbmini"
-			elif procmodel.endswith("1000de"):
-				brand = "Golden Interstar"
-				model = "Xpeed LX"
-			elif procmodel.endswith("9000de"):
-				brand = "Golden Interstar"
-				model = "Xpeed LX3"
-			elif procmodel.endswith("1000lx"):
-				brand = "Golden Interstar"
-				model = "Xpeed LX"
-			elif procmodel.endswith("de"):
-				brand = "Golden Interstar"
-			elif procmodel.endswith("1000am"):
-				brand = "Atemio"
-				model = "5x00"
-			else:
-				brand = "Venton"
-				model = "HDx"
-		elif procmodel.startswith("unibox-"):
-			brand = "Venton"
-			model = "HDe"
-		elif procmodel == "hd1100":
-			brand = "Mut@nt"
-			model = "HD1100"
-		elif procmodel == "hd1200":
-			brand = "Mut@nt"
-			model = "HD1200"
-		elif procmodel == "hd1265":
-			brand = "Mut@nt"
-			model = "HD1265"
-		elif procmodel == "hd2400":
-			brand = "Mut@nt"
-			model = "HD2400"
-		elif procmodel == "hd51":
-			brand = "Mut@nt"
-			model = "HD51"
-			grabpip = 1
-		elif procmodel == "hd11":
-			brand = "Mut@nt"
-			model = "HD11"
-		elif procmodel == "hd500c":
-			brand = "Mut@nt"
-			model = "HD500c"
-		elif procmodel == "hd530c":
-			brand = "Mut@nt"
-			model = "HD530c"
-		elif procmodel == "hd60":
-			brand = "Mut@nt"
-			model = "HD60"
-		elif procmodel == "hd66se":
-			brand = "Mut@nt"
-			model = "HD66se"
-		elif procmodel == "multibox":
-			brand = "MaXytec"
-			model = "Multibox"
-			grabpip = 1
-		elif procmodel == "multiboxse":
-			brand = "MaXitec"
-			model = "MultiboxSE"
-			grabpip = 1
-		elif procmodel == "arivalink200":
+		if procmodel == "arivalink200":
 			brand = "Ferguson"
 			model = "Ariva @Link 200"
 		elif procmodel.startswith("spark"):
@@ -357,204 +191,6 @@
 				model = "Spark 7162"
 			else:
 				model = "Spark"
-		elif procmodel == "spycat":
-			brand = "Spycat"
-			model = "Spycat"
-		elif procmodel == "spycatmini":
-			brand = "Spycat"
-			model = "Spycat Mini"
-		elif procmodel == "spycatminiplus":
-			brand = "Spycat"
-			model = "Spycat Mini+"
-		elif procmodel == "spycat4kmini":
-			brand = "Spycat"
-			model = "spycat 4K Mini"
-		elif procmodel == "vipercombo":
-			brand = "Amiko"
-			model = "ViperCombo"
-		elif procmodel == "vipert2c":
-			brand = "Amiko"
-			model = "ViperT2C"
-		elif procmodel == "vipercombohdd":
-			brand = "Amiko"
-			model = "ViperComboHDD"
-		elif procmodel == "viperslim":
-			brand = "Amiko"
-			model = "Viper Slim"
-		elif procmodel == "wetekplay":
-			brand = "WeTeK"
-			model = "Play"
-		elif procmodel.startswith("os"):
-			brand = "Edision"
-			if procmodel == "osmini":
-				model = "OS Mini"
-			elif procmodel == "osminiplus":
-				model = "OS Mini+"
-			elif procmodel == "osmega":
-				model = "OS Mega"
-			elif procmodel == "osnino":
-				model = "OS Nino"
-			elif procmodel == "osninoplus":
-				model = "OS Nino+"
-			elif procmodel == "osninopro":
-				model = "OS Nino Pro"
-			elif procmodel == "osmio4k":
-				model = "OS Mio 4K"
-				grabpip = 1
-			elif procmodel == "osmio4kplus":
-				model = "OS Mio 4K+"
-				grabpip = 1
-			elif procmodel == "osmini4k":
-				model = "OS Mini 4K"
-				grabpip = 1
-			else:
-				model = procmodel
-		elif procmodel == "h3":
-			brand = "Zgemma"
-			model = "H3 series"
-		elif procmodel == "h4":
-			brand = "Zgemma"
-			model = "H4 series"
-		elif procmodel == "h5":
-			brand = "Zgemma"
-			model = "H5 series"
-		elif procmodel == "h6":
-			brand = "Zgemma"
-			model = "H6 series"
-		elif procmodel == "h7":
-			brand = "Zgemma"
-			model = "H7 series"
-			grabpip = 1
-		elif procmodel == "h8":
-			brand = "Zgemma"
-			model = "H8 series"
-		elif procmodel == "h9":
-			brand = "Zgemma"
-			model = "H9 series"
-			grabpip = 1
-		elif procmodel == "lc":
-			brand = "Zgemma"
-			model = "LC"
-		elif procmodel == "sh1":
-			brand = "Zgemma"
-			model = "Star series"
-		elif procmodel == "i55":
-			brand = "Zgemma"
-			model = "i55"
-		elif procmodel == "i55plus":
-			brand = "Zgemma"
-			model = "i55Plus"
-			grabpip = 1
-		elif procmodel == "hzero":
-			brand = "Zgemma"
-			model = "Hzero"
-		elif procmodel == "h8.2h":
-			brand = "Zgemma"
-			model = "H8.2H"
-		elif procmodel == "h9.s":
-			brand = "Zgemma"
-			model = "H9.S"
-			grabpip = 1
-		elif procmodel == "h9.t":
-			brand = "Zgemma"
-			model = "H9.T"
-			grabpip = 1
-		elif procmodel == "h9.2h":
-			brand = "Zgemma"
-			model = "H9.2H"
-			grabpip = 1
-		elif procmodel == "h9.2s":
-			brand = "Zgemma"
-			model = "H9.2S"
-			grabpip = 1
-		elif procmodel == "h9 combo" or procmodel == "h9combo":
-			procmodel = "h9combo"
-			brand = "Zgemma"
-			model = "H9 Combo"
-			grabpip = 1
-		elif procmodel == "h9combose":
-			procmodel = "h9combose"
-			brand = "Zgemma"
-			model = "H9 Combo SE"
-			grabpip = 1
-		elif procmodel == "h9 twin" or procmodel == "h9twin":
-			procmodel = "h9twin"
-			brand = "Zgemma"
-			model = "H9 Twin"
-			grabpip = 1
-		elif procmodel == "h9twinse":
-			procmodel = "h9twinse"
-			brand = "Zgemma"
-			model = "H9 Twin SE"
-			grabpip = 1
-		elif procmodel == "h9.s.se":
-			brand = "Zgemma"
-			model = "H9.S SE"
-			grabpip = 1
-		elif procmodel == "h9.2s.se":
-			brand = "Zgemma"
-			model = "H9.2S SE"
-			grabpip = 1
-		elif procmodel == "h9.2h.se":
-			brand = "Zgemma"
-			model = "H9.2H SE"
-			grabpip = 1
-		elif procmodel == "h10.t":
-			brand = "Zgemma"
-			model = "H10 Combo"
-			grabpip = 1
-		elif procmodel == "h10.2s":
-			brand = "Zgemma"
-			model = "H10.2S"
-			grabpip = 1
-		elif procmodel == "h10.2h":
-			brand = "Zgemma"
-			model = "H10.2H"
-			grabpip = 1
-		elif procmodel == "h11":
-			brand = "Zgemma"
-			model = "H11"
-			grabpip = 1
-		elif procmodel == "h11.s":
-			brand = "Zgemma"
-			model = "H11.S"
-			grabpip = 1
-		elif procmodel == "vs1500":
-			brand = "Vimastec"
-			model = "vs1500"
-			grabpip = 1
-		elif procmodel == "sf8008m":
-			brand = "Octagon"
-			model = "SF8008 4K Mini"
-		elif procmodel.startswith("sf"):
-			brand = "Octagon"
-			if procmodel.startswith("sf8008"):
-				sf8008type = open("/proc/stb/info/type").read()
-				if sf8008type.startswith("11"):
-					procmodel = "sf8008t"
-					model = "SF8008 4K Twin"
-				elif sf8008type.startswith("12"):
-					procmodel = "sf8008c"
-					model = "SF8008 4K Combo"
-				else:  # sf8008type.startswith("10")
-					procmodel = "sf8008s"
-					model = "SF8008 4K Single"
-			else:
-				model = procmodel.upper()
-		elif procmodel == "e4hd":
-			brand = "Axas"
-			model = "E4HD"
-			lcd = 1
-			grabpip = 1
-		elif procmodel == "ustym4kpro":
-			brand = "uClan"
-			model = "Ustym 4K Pro"
-		elif procmodel == "pulse4k":
-			brand = "AB-COM"
-			model = "PULSe 4K"
-		elif procmodel == "pulse4kmini":
-			brand = "AB-COM"
-			model = "PULSe 4K MINI"
 		else:
 			model = procmodel
 
@@ -650,20 +286,29 @@
 	info['type'] = type
 
 	remote = "dmm1"
-	if procmodel in ("vusolo", "vuduo", "vuuno", "vusolo2", "vusolose", "vuzero", "vusolo4k", "vuuno4k", "vuultimo4k"):
-		remote = "vu_normal"
-	elif procmodel == "vuduo2":
-		remote = "vu_duo2"
-	elif procmodel == "vuultimo":
-		remote = "vu_ultimo"
-	elif procmodel in ("vuuno4kse", "vuzero4k", "vuduo4k", "vuduo4kse"):
-		remote = "vu_normal_02"
-	elif procmodel == "e3hd":
-		remote = "e3hd"
-	elif procmodel in ("et9x00", "et9000", "et9100", "et9200", "et9500"):
-		remote = "et9x00"
-	elif procmodel in ("et5x00", "et5000", "et6x00", "et6000"):
-		remote = "et5x00"
+	
+	if procmodel in ("fs9000", "hs8200"):
+		remote = "fs9000"
+	if procmodel in ("hs9510", "hs7420", "hs7429", "hs7810a", "hs7819"):
+		remote = "hs9510"
+	if procmodel in ("hs7110", "hs7119"):
+		remote = "hs7110"
+	elif procmodel == "ufs910":
+		remote = "ufs910"
+	elif procmodel in ("ufs912", "ufs922"):
+		remote = "ufs912"
+	elif procmodel == "ufs913":
+		remote = "ufs913"
+	elif procmodel == "adb_box":
+		remote = "adb_xmp"
+	elif procmodel == "hl101":
+		remote = "hl101"
+	elif procmodel in ("vip1-v1", "vip1-v2", "vip2"):
+		remote = "vip_1"
+	elif procmodel in ("cuberevo", "cuberevo-mini", "cuberevo-mini2", "cuberevo-2000hd", "cuberevo-3000hd", "cuberevo-9500hd"):
+		remote = "cuberevo_uni"
+	elif procmodel in ("cuberevo-250hd", "cuberevo-mini-fta"):
+		remote = "cuberevo"
 	elif procmodel in ("et4x00", "et4000"):
 		remote = "et4x00"
 	elif procmodel == "et6500":
@@ -833,7 +478,7 @@
 	else:
 		# OE 2.2 uses apt, not opkg
 		if not fileExists("/etc/opkg/all-feed.conf"):
-			oever = "OE 2.2"
+			oever = "Audioniek's buildsystem"
 		else:
 			try:
 				f = open("/etc/opkg/all-feed.conf", 'r')

