--- enigma2-servicemp3epl-0.1/servicemp3/servicemp3.h.org
+++ enigma2-servicemp3epl-0.1/servicemp3/servicemp3.h
@@ -182,7 +182,6 @@
 	RESULT frontendInfo(ePtr<iFrontendInformation> &ptr) { ptr = 0; return -1; }
 	RESULT subServices(ePtr<iSubserviceList> &ptr) { ptr = 0; return -1; }
 	RESULT timeshift(ePtr<iTimeshiftService> &ptr) { ptr = 0; return -1; }
-	RESULT tap(ePtr<iTapService> &ptr) { ptr = nullptr; return -1; };
 //	RESULT cueSheet(ePtr<iCueSheet> &ptr) { ptr = 0; return -1; }
 
 		// iCueSheet
--- enigma2-servicemp3epl-0.1/servicemp3/servicelibpl.h.org
+++ enigma2-servicemp3epl-0.1/servicemp3/servicelibpl.h
@@ -115,7 +115,6 @@
 	RESULT frontendInfo(ePtr<iFrontendInformation> &ptr) { ptr = 0; return -1; }
 	RESULT subServices(ePtr<iSubserviceList> &ptr) { ptr = 0; return -1; }
 	RESULT timeshift(ePtr<iTimeshiftService> &ptr) { ptr = 0; return -1; }
-	RESULT tap(ePtr<iTapService> &ptr) { ptr = nullptr; return -1; };
 
 	// iCueSheet
 	PyObject *getCutList();

