--- a/include/linux/dvb/ca.h.org
+++ b/include/linux/dvb/ca.h
@@ -88,6 +88,13 @@
 	CA_PARITY_ODD
 };
 
+enum ca_descr_algo
+{
+	CA_ALGO_DVBCSA,
+	CA_ALGO_DES,
+	CA_ALGO_AES128,
+};
+
 typedef struct ca_descr_data {
 	unsigned int index;
 	enum ca_descr_parity parity;
@@ -95,6 +102,13 @@
 	unsigned int length;
 	unsigned char *data;
 }ca_descr_data_t;
+
+typedef struct ca_descr_mode {
+	unsigned int index;
+	enum ca_descr_algo algo;
+	int session_handle;
+	int session_descrabler_handle;
+} ca_descr_mode_t;   
 
 #define CA_RESET          _IO('o', 128)
 #define CA_GET_CAP        _IOR('o', 129, ca_caps_t)
@@ -104,6 +118,7 @@
 #define CA_SEND_MSG       _IOW('o', 133, ca_msg_t)
 #define CA_SET_DESCR      _IOW('o', 134, ca_descr_t)
 #define CA_SET_PID        _IOW('o', 135, ca_pid_t)
+#define CA_SET_DESCR_MODE _IOW('o', 136, ca_descr_mode_t)
 #define CA_SET_DESCR_DATA _IOW('o', 137, ca_descr_data_t)
 
 #endif

