--- a/config/kernel/linux.in.org
+++ b/config/kernel/linux.in
@@ -284,6 +284,7 @@
     default "2.6.35.9" if KERNEL_V_2_6_35
     default "2.6.34.7" if KERNEL_V_2_6_34
     default "2.6.33.7" if KERNEL_V_2_6_33
-    default "2.6.32.27" if KERNEL_V_2_6_32
+#    default "2.6.32.27" if KERNEL_V_2_6_32
+    default "2.6.32.71" if KERNEL_V_2_6_32
 
 endif # ! KERNEL_LINUX_CUSTOM
#--- a/config/arch/sh.in.org
#+++ b/config/arch/sh.in
#@@ -5,7 +5,6 @@
# ## select ARCH_USE_MMU
# ## select ARCH_SUPPORTS_BOTH_ENDIAN
# ## select ARCH_DEFAULT_LE
#-## select ARCH_REQUIRES_MULTILIB
# ##
# ## help The Super-H architecture, as defined by:
# ## help     http://www.renesas.com/fmwk.jsp?cnt=superh_family_landing.jsp&fp=/products/mpumcu/superh_family/

