--- a/config/cc/gcc.in.org
+++ b/config/cc/gcc.in
@@ -48,9 +48,9 @@
     depends on CC_GCC_SHOW_LINARO
     select CC_GCC_4_9
 
-config CC_GCC_V_4_9_3
-    bool
-    prompt "4.9.3"
+config CC_GCC_V_4_9_4
+    bool
+    prompt "4.9.4"
     select CC_GCC_4_9
 
 config CC_GCC_V_linaro_4_8
@@ -369,7 +369,7 @@
 # CT_INSERT_VERSION_STRING_BELOW
     default "5.2.0" if CC_GCC_V_5_2_0
     default "linaro-4.9-2015.06" if CC_GCC_V_linaro_4_9
-    default "4.9.3" if CC_GCC_V_4_9_3
+    default "4.9.4" if CC_GCC_V_4_9_4
     default "linaro-4.8-2015.06" if CC_GCC_V_linaro_4_8
     default "4.8.5" if CC_GCC_V_4_8_5
     default "linaro-4.7-2014.06" if CC_GCC_V_linaro_4_7
--- a/samples/sh4-unknown-linux-gnu/crosstool.config.org
+++ b/samples/sh4-unknown-linux-gnu/crosstool.config
@@ -5,7 +5,7 @@
 CT_ARCH_SH_SH4=y
 CT_KERNEL_linux=y
 CT_BINUTILS_PLUGINS=y
-CT_CC_GCC_V_4_9_3=y
+CT_CC_GCC_V_4_9_4=y
 CT_CC_LANG_CXX=y
 CT_DEBUG_gdb=y
 # CT_GDB_CROSS_PYTHON is not set

