# Commit 154632ace0e0a69619e7d7b2b72bb2de1adbcd8f 25/10/2020
--- neutrino-ddt.org/acinclude.m4
+++ neutrino-ddt/acinclude.m4
@@ -284,45 +284,13 @@
 AC_ARG_WITH(boxtype,
 	AS_HELP_STRING([--with-boxtype], [valid values: spark, generic, armbox, duckbox, spark7162, mipsbox]),
 	[case "${withval}" in
-		generic|armbox)
+		generic|armbox|duckbox)
 			BOXTYPE="$withval"
 		;;
 		spark|spark7162)
 			BOXTYPE="spark"
 			BOXMODEL="$withval"
 		;;
-		ufs*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		atevio*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		fortis*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		octagon*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		hs7*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		cuberevo*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		ipbox*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		tf*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
 		hd51|hd60|hd61|bre2ze4k|vusolo4k|vuduo4k|vuduo4kse|vuultimo4k|vuzero4k|vuuno4kse|vuuno4k|h7|osmio4k|osmio4kplus)
 			BOXTYPE="armbox"
 			BOXMODEL="$withval"
@@ -338,13 +306,13 @@
 	[BOXTYPE="generic"])
 
 AC_ARG_WITH(boxmodel,
-	AS_HELP_STRING([--with-boxmodel], [valid for duckbox: ufs910, ufs912, ufs913, ufs922, atevio7500, fortis_hdbox, octagon1008, cuberevo, cuberevo_mini, cuberevo_mini2, cuberevo_250hd, cuberevo_2000hd, cuberevo_3000hd, ipbox9900, ipbox99, ipbox55, tf7700])
+	AS_HELP_STRING([--with-boxmodel], [valid for duckbox: ufs910, ufs912, ufs913, ufs922, fs9000, hs7110, hs7420, hs7810a, hs7119, hs7429, hs7819, hs8200, hs9510, cuberevo, cuberevo_mini, cuberevo_mini2, cuberevo_mini_fta, cuberevo_250hd, cuberevo_2000hd, cuberevo_3000hd, ipbox9900, ipbox99, ipbox55, tf7700, hl101, adb_box, vip1_v1, vip1_v2, vip2, vitamin_hd5000, pace7241, adb2850, opt9600, opt9600mini, atemio520])
 AS_HELP_STRING([], [valid for spark: spark, spark7162])
 AS_HELP_STRING([], [valid for armbox: bre2ze4k, hd51, vusolo4k, vuduo4k, vuduo4kse, vuultimo4k, vuzero4k, vuuno4kse, vuuno4k, h7])
 AS_HELP_STRING([], [valid for mipsbox: vuduo, dm8000])
 AS_HELP_STRING([], [valid for generic: raspi]),
 	[case "${withval}" in
-		ufs910|ufs912|ufs913|ufs922|atevio7500|fortis_hdbox|octagon1008|cuberevo|cuberevo_mini|cuberevo_mini2|cuberevo_250hd|cuberevo_2000hd|cuberevo_3000hd|ipbox9900|ipbox99|ipbox55|tf7700)
+		ufs910|ufs912|ufs913|ufs922|fs9000|hs7110|hs7420|hs7810a|hs7119|hs7429|hs7819|hs8200|hs9510|cuberevo|cuberevo_mini|cuberevo_mini2|cuberevo_mini_fta|cuberevo_250hd|cuberevo_2000hd|cuberevo_3000hd|ipbox9900|ipbox99|ipbox55|tf7700|hl101|adb_box|vip1_v1|vip1_v2|vip2|vitamin_hd5000|pace7241|adb2850|opt9600|opt9600mini|atemio520)
 			if test "$BOXTYPE" = "duckbox"; then
 				BOXMODEL="$withval"
 			else
@@ -399,13 +367,21 @@
 AM_CONDITIONAL(BOXMODEL_UFS922, test "$BOXMODEL" = "ufs922")
 AM_CONDITIONAL(BOXMODEL_SPARK, test "$BOXMODEL" = "spark")
 AM_CONDITIONAL(BOXMODEL_SPARK7162, test "$BOXMODEL" = "spark7162")
-AM_CONDITIONAL(BOXMODEL_ATEVIO7500, test "$BOXMODEL" = "atevio7500")
-AM_CONDITIONAL(BOXMODEL_FORTIS_HDBOX, test "$BOXMODEL" = "fortis_hdbox")
-AM_CONDITIONAL(BOXMODEL_OCTAGON1008, test "$BOXMODEL" = "octagon1008")
+AM_CONDITIONAL(BOXMODEL_FS9000, test "$BOXMODEL" = "fs9000")
+AM_CONDITIONAL(BOXMODEL_HS8200, test "$BOXMODEL" = "hs8200")
+AM_CONDITIONAL(BOXMODEL_HS7110, test "$BOXMODEL" = "hs7110")
+AM_CONDITIONAL(BOXMODEL_HS7420, test "$BOXMODEL" = "hs7420")
+AM_CONDITIONAL(BOXMODEL_HS7810A, test "$BOXMODEL" = "hs7810a")
+AM_CONDITIONAL(BOXMODEL_HS7119, test "$BOXMODEL" = "hs7119")
+AM_CONDITIONAL(BOXMODEL_HS7429, test "$BOXMODEL" = "hs7429")
+AM_CONDITIONAL(BOXMODEL_HS7819, test "$BOXMODEL" = "hs7819")
+AM_CONDITIONAL(BOXMODEL_HS9510, test "$BOXMODEL" = "hs9510")
+AM_CONDITIONAL(BOXMODEL_DP7000, test "$BOXMODEL" = "dp7000")
 
 AM_CONDITIONAL(BOXMODEL_CUBEREVO, test "$BOXMODEL" = "cuberevo")
 AM_CONDITIONAL(BOXMODEL_CUBEREVO_MINI, test "$BOXMODEL" = "cuberevo_mini")
 AM_CONDITIONAL(BOXMODEL_CUBEREVO_MINI2, test "$BOXMODEL" = "cuberevo_mini2")
+AM_CONDITIONAL(BOXMODEL_CUBEREVO_200HD, test "$BOXMODEL" = "cuberevo_mini_fta")
 AM_CONDITIONAL(BOXMODEL_CUBEREVO_250HD, test "$BOXMODEL" = "cuberevo_250hd")
 AM_CONDITIONAL(BOXMODEL_CUBEREVO_2000HD, test "$BOXMODEL" = "cuberevo_2000hd")
 AM_CONDITIONAL(BOXMODEL_CUBEREVO_3000HD, test "$BOXMODEL" = "cuberevo_3000hd")
@@ -413,6 +389,17 @@
 AM_CONDITIONAL(BOXMODEL_IPBOX99, test "$BOXMODEL" = "ipbox99")
 AM_CONDITIONAL(BOXMODEL_IPBOX55, test "$BOXMODEL" = "ipbox55")
 AM_CONDITIONAL(BOXMODEL_TF7700, test "$BOXMODEL" = "tf7700")
+AM_CONDITIONAL(BOXMODEL_HL101, test "$BOXMODEL" = "hl101")
+AM_CONDITIONAL(BOXMODEL_VIP1_V1, test "$BOXMODEL" = "vip1_v1")
+AM_CONDITIONAL(BOXMODEL_VIP1_V2, test "$BOXMODEL" = "vip1_v2")
+AM_CONDITIONAL(BOXMODEL_VIP2, test "$BOXMODEL" = "vip2")
+AM_CONDITIONAL(BOXMODEL_ADB_BOX, test "$BOXMODEL" = "adb_box")
+AM_CONDITIONAL(BOXMODEL_VITAMIN_HD5000, test "$BOXMODEL" = "vitamin_hd5000")
+AM_CONDITIONAL(BOXMODEL_PACE7241, test "$BOXMODEL" = "pace7241")
+AM_CONDITIONAL(BOXMODEL_ADB_2850, test "$BOXMODEL" = "adb2850")
+AM_CONDITIONAL(BOXMODEL_OPT9600, test "$BOXMODEL" = "opt9600")
+AM_CONDITIONAL(BOXMODEL_OPT9600MINI, test "$BOXMODEL" = "opt9600mini")
+AM_CONDITIONAL(BOXMODEL_ATEMIO520, test "$BOXMODEL" = "atemio520")
 
 AM_CONDITIONAL(BOXMODEL_HD51, test "$BOXMODEL" = "hd51")
 AM_CONDITIONAL(BOXMODEL_HD60, test "$BOXMODEL" = "hd60")
@@ -464,24 +451,38 @@
 	AC_DEFINE(BOXMODEL_SPARK, 1, [spark])
 elif test "$BOXMODEL" = "spark7162"; then
 	AC_DEFINE(BOXMODEL_SPARK7162, 1, [spark7162])
-elif test "$BOXMODEL" = "atevio7500"; then
-	AC_DEFINE(BOXMODEL_ATEVIO7500, 1, [atevio7500])
-elif test "$BOXMODEL" = "fortis_hdbox"; then
-	AC_DEFINE(BOXMODEL_FORTIS_HDBOX, 1, [fortis_hdbox])
-elif test "$BOXMODEL" = "octagon1008"; then
-	AC_DEFINE(BOXMODEL_OCTAGON1008, 1, [octagon1008])
+elif test "$BOXMODEL" = "hs8200"; then
+	AC_DEFINE(BOXMODEL_HS8200, 1, [hs8200])
+elif test "$BOXMODEL" = "fs9000"; then
+	AC_DEFINE(BOXMODEL_FS9000, 1, [fs9000])
+elif test "$BOXMODEL" = "hs9510"; then
+	AC_DEFINE(BOXMODEL_HS9510, 1, [hs9510])
+elif test "$BOXMODEL" = "hs7110"; then
+	AC_DEFINE(BOXMODEL_HS7110, 1, [hs7110])
+elif test "$BOXMODEL" = "hs7420"; then
+	AC_DEFINE(BOXMODEL_HS7420, 1, [hs7420])
+elif test "$BOXMODEL" = "hs7810a"; then
+	AC_DEFINE(BOXMODEL_HS7810A, 1, [hs7810a])
+elif test "$BOXMODEL" = "hs7119"; then
+	AC_DEFINE(BOXMODEL_HS7119, 1, [hs7119])
+elif test "$BOXMODEL" = "hs7429"; then
+	AC_DEFINE(BOXMODEL_HS7429, 1, [hs7429])
+elif test "$BOXMODEL" = "hs7819"; then
+	AC_DEFINE(BOXMODEL_HS7819, 1, [hs7819])
 elif test "$BOXMODEL" = "cuberevo"; then
 	AC_DEFINE(BOXMODEL_CUBEREVO, 1, [cuberevo])
 elif test "$BOXMODEL" = "cuberevo_mini"; then
-	AC_DEFINE(BOXMODEL_CUBEREVO_MINI, 1, [cuberevo_mini])
+	AC_DEFINE(BOXMODEL_CUBEREVO_MINI, 1, [cuberevo-mini])
 elif test "$BOXMODEL" = "cuberevo_mini2"; then
-	AC_DEFINE(BOXMODEL_CUBEREVO_MINI2, 1, [cuberevo_mini2])
+	AC_DEFINE(BOXMODEL_CUBEREVO_MINI2, 1, [cuberevo-mini2])
+elif test "$BOXMODEL" = "cuberevo_mini_fta"; then
+	AC_DEFINE(BOXMODEL_CUBEREVO_MINI_FTA, 1, [cuberevo-mini-fta])
 elif test "$BOXMODEL" = "cuberevo_250hd"; then
-	AC_DEFINE(BOXMODEL_CUBEREVO_250HD, 1, [cuberevo_250hd])
+	AC_DEFINE(BOXMODEL_CUBEREVO_250HD, 1, [cuberevo-250hd])
 elif test "$BOXMODEL" = "cuberevo_2000hd"; then
-	AC_DEFINE(BOXMODEL_CUBEREVO_2000HD, 1, [cuberevo_2000hd])
+	AC_DEFINE(BOXMODEL_CUBEREVO_2000HD, 1, [cuberevo-2000hd])
 elif test "$BOXMODEL" = "cuberevo_3000hd"; then
-	AC_DEFINE(BOXMODEL_CUBEREVO_3000HD, 1, [cuberevo_3000hd])
+	AC_DEFINE(BOXMODEL_CUBEREVO_3000HD, 1, [cuberevo-3000hd])
 elif test "$BOXMODEL" = "ipbox9900"; then
 	AC_DEFINE(BOXMODEL_IPBOX9900, 1, [ipbox9900])
 elif test "$BOXMODEL" = "ipbox99"; then
@@ -490,6 +491,29 @@
 	AC_DEFINE(BOXMODEL_IPBOX55, 1, [ipbox55])
 elif test "$BOXMODEL" = "tf7700"; then
 	AC_DEFINE(BOXMODEL_TF7700, 1, [tf7700])
+ elif test "$BOXMODEL" = "hl101"; then
+ 	AC_DEFINE(BOXMODEL_HL101, 1, [hl101])
+ elif test "$BOXMODEL" = "vip1_v1"; then
+ 	AC_DEFINE(BOXMODEL_VIP1_V1, 1, [vip1-v1])
+ elif test "$BOXMODEL" = "vip1_v2"; then
+ 	AC_DEFINE(BOXMODEL_VIP1_V2, 1, [vip1-v2])
+ elif test "$BOXMODEL" = "vip2"; then
+ 	AC_DEFINE(BOXMODEL_VIP2, 1, [vip2])
+elif test "$BOXMODEL" = "adb_box"; then
+	AC_DEFINE(BOXMODEL_ADB_BOX, 1, [adb-box])
+elif test "$BOXMODEL" = "vitamin_hd5000"; then
+	AC_DEFINE(BOXMODEL_VITAMIN_HD5000, 1, [vitamin-hd5000])
+elif test "$BOXMODEL" = "pace7241"; then
+	AC_DEFINE(BOXMODEL_PACE7241, 1, [pace7241])
+elif test "$BOXMODEL" = "adb2850"; then
+	AC_DEFINE(BOXMODEL_ADB_2850, 1, [adb2850])
+elif test "$BOXMODEL" = "opt9600"; then
+	AC_DEFINE(BOXMODEL_OPT9600, 1, [opt9600])
+elif test "$BOXMODEL" = "opt9600mini"; then
+	AC_DEFINE(BOXMODEL_OPT9600MINI, 1, [opt9600mini])
+elif test "$BOXMODEL" = "atemio520"; then
+	AC_DEFINE(BOXMODEL_ATEMIO520, 1, [atemio520])
+
 elif test "$BOXMODEL" = "bre2ze4k"; then
 	AC_DEFINE(BOXMODEL_BRE2ZE4K, 1, [bre2ze4k])
 elif test "$BOXMODEL" = "hd51"; then
--- neutrino-ddt.org/src/driver/fbaccel.cpp
+++ neutrino-ddt/src/driver/fbaccel.cpp
@@ -108,7 +108,7 @@
 	}
 	backbuf_sz = 1280 * 720 * sizeof(fb_pixel_t);
 	BPAMemAllocMemData bpa_data;
-#if BOXMODEL_OCTAGON1008 || BOXMODEL_FORTIS_HDBOX || BOXMODEL_CUBEREVO || BOXMODEL_CUBEREVO_MINI || BOXMODEL_CUBEREVO_MINI2 || BOXMODEL_CUBEREVO_250HD || BOXMODEL_CUBEREVO_2000HD || BOXMODEL_CUBEREVO_3000HD || BOXMODEL_IPBOX9900 || BOXMODEL_IPBOX99 || BOXMODEL_IPBOX55 || BOXMODEL_TF7700
+#if BOXMODEL_HS9510 || BOXMODEL_FS9000 || BOXMODEL_CUBEREVO || BOXMODEL_CUBEREVO_MINI || BOXMODEL_CUBEREVO_MINI2 || BOXMODEL_CUBEREVO_250HD || BOXMODEL_CUBEREVO_MINI_FTA|| BOXMODEL_CUBEREVO_2000HD || BOXMODEL_CUBEREVO_3000HD || BOXMODEL_IPBOX9900 || BOXMODEL_IPBOX99 || BOXMODEL_IPBOX55 || BOXMODEL_TF7700 || BOXMODEL_OPT9600
 	bpa_data.bpa_part = (char *)"LMI_SYS";
 #else
 	bpa_data.bpa_part = (char *)"LMI_VID";
--- neutrino-ddt.org/src/driver/fb_accel_sti_ddt.cpp
+++ neutrino-ddt/src/driver/fb_accel_sti_ddt.cpp
@@ -96,7 +96,7 @@
 	}
 	backbuf_sz = DEFAULT_XRES * DEFAULT_YRES * sizeof(fb_pixel_t);
 	BPAMemAllocMemData bpa_data;
-#if BOXMODEL_OCTAGON1008 || BOXMODEL_FORTIS_HDBOX || BOXMODEL_CUBEREVO || BOXMODEL_CUBEREVO_MINI || BOXMODEL_CUBEREVO_MINI2 || BOXMODEL_CUBEREVO_250HD || BOXMODEL_CUBEREVO_2000HD || BOXMODEL_CUBEREVO_3000HD || BOXMODEL_IPBOX9900 || BOXMODEL_IPBOX99 || BOXMODEL_IPBOX55 || BOXMODEL_TF7700
+#if BOXMODEL_HS9510 || BOXMODEL_FS9000 || BOXMODEL_CUBEREVO || BOXMODEL_CUBEREVO_MINI || BOXMODEL_CUBEREVO_MINI2 || BOXMODEL_CUBEREVO_250HD || BOXMODEL_CUBEREVO_MINI_FTA || BOXMODEL_CUBEREVO_2000HD || BOXMODEL_CUBEREVO_3000HD || BOXMODEL_IPBOX9900 || BOXMODEL_IPBOX99 || BOXMODEL_IPBOX55 || BOXMODEL_TF7700 || BOXMODEL_OPT9600
 	bpa_data.bpa_part = (char *)"LMI_SYS";
 #else
 	bpa_data.bpa_part = (char *)"LMI_VID";
--- neutrino-ddt.org/src/driver/record.cpp
+++ neutrino-ddt/src/driver/record.cpp
@@ -1408,7 +1408,7 @@
 		autoshift = false;
 #ifdef HAVE_SPARK_HARDWARE
 	CVFD::getInstance()->SetIcons(SPARK_TIMESHIFT, false);
-#elif defined(BOXMODEL_FORTIS_HDBOX)
+#elif defined(BOXMODEL_FS9000)
 	CVFD::getInstance()->ShowIcon(FP_ICON_TIMESHIFT, false);
 #endif
 
@@ -1573,7 +1573,7 @@
 		t_channel_id live_channel_id = CZapit::getInstance()->GetCurrentChannelID();
 #ifdef HAVE_SPARK_HARDWARE
 		CVFD::getInstance()->SetIcons(SPARK_TIMESHIFT, true);
-#elif defined(BOXMODEL_FORTIS_HDBOX)
+#elif defined(BOXMODEL_FS9000)
 		CVFD::getInstance()->ShowIcon(FP_ICON_TIMESHIFT, true);
 #endif
 		bool tstarted = false;
--- neutrino-ddt.org/src/neutrino.cpp
+++ neutrino-ddt/src/neutrino.cpp
@@ -3097,6 +3097,7 @@
 					switchTvRadioMode(); //used with defined default tv/radio rc key
 			}
 			/* in case key_subchannel_up/down redefined */
+#if !defined(BOXMODEL_OPT9600)
 			else if( msg == CRCInput::RC_left || msg == CRCInput::RC_right) {
 				switch (g_settings.mode_left_right_key_tv)
 				{
@@ -3113,6 +3114,7 @@
 						break;
 				}
 			}
+#endif
 			else if( msg == CRCInput::RC_epg ) {
 				InfoClock->enableInfoClock(false);
 				StopSubtitles();
@@ -3404,6 +3406,10 @@
 #if defined (BOXMODEL_IPBOX9900) || defined (BOXMODEL_IPBOX99)
 		system("echo 0 > /proc/stb/misc/fan");
 #endif
+#if defined (BOXMODEL_CUBEREVO) || defined (BOXMODEL_CUBEREVO_9500HD)
+		system("echo 1 > /proc/stb/fan");  // switch fan on
+		system("echo 128 > /proc/stb/fan_pwm");
+#endif
 	}
 }
 
@@ -3664,7 +3670,11 @@
 			}
 #endif
 	}
+#if defined(BOXMODEL_OPT9600)
+	else if ((msg == CRCInput::RC_left) || (msg == CRCInput::RC_right))
+#else
 	else if ((msg == CRCInput::RC_plus) || (msg == CRCInput::RC_minus))
+#endif
 	{
 		g_volume->setVolume(msg);
 #if HAVE_DUCKBOX_HARDWARE
--- neutrino-ddt.org/src/driver/vfd.cpp
+++ neutrino-ddt/src/driver/vfd.cpp
@@ -52,19 +52,42 @@
 #include <stropts.h>
 #define VFD_DEVICE "/dev/vfd"
 
-#if defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_TF7700)
+#if defined (BOXMODEL_HS9510) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_TF7700) \
+ || defined (BOXMODEL_OPT9600)
 #define VFDLENGTH 8
-#elif defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#elif defined (BOXMODEL_FS9000) \
+ ||   defined (BOXMODEL_HS8200) \
+ ||   defined (BOXMODEL_CUBEREVO) \
+ ||   defined (BOXMODEL_CUBEREVO_9500HD)
 #define VFDLENGTH 12
-#elif  defined (BOXMODEL_CUBEREVO_250HD) || defined (BOXMODEL_IPBOX55)
+#elif defined (BOXMODEL_HS7810A) \
+ ||   defined (BOXMODEL_HS7119) \
+ ||   defined (BOXMODEL_HS7819) \
+ ||   defined (BOXMODEL_CUBEREVO_250HD) \
+ ||   defined (BOXMODEL_CUBEREVO_MINI_FTA) \
+ ||   defined (BOXMODEL_IPBOX55) \
+ ||   defined (BOXMODEL_ATEMIO520) \
+ ||   defined (BOXMODEL_OPT9600MINI)
 #define VFDLENGTH 4
-#elif defined (BOXMODEL_IPBOX9900) || defined (BOXMODEL_IPBOX99)
+#elif defined (BOXMODEL_HS7110)
+#define VFDLENGTH 0
+#elif defined (BOXMODEL_IPBOX9900) \
+ ||   defined (BOXMODEL_IPBOX99) \
+ ||   defined (BOXMODEL_CUBEREVO_MINI) \
+ ||   defined (BOXMODEL_CUBEREVO_MINI2) \
+ ||   defined (BOXMODEL_CUBEREVO_2000HD) \
+ ||   defined (BOXMODEL_CUBEREVO_3000HD)
 #define VFDLENGTH 14
+#elif defined (BOXMODEL_ADB_BOX)
+#define VFDLENGTH 15
 #else
 #define VFDLENGTH 16
 #endif
 
-#define SCROLL_TIME 100000
+#define SCROLL_TIME 350000
 
 bool invert = false;
 char g_str[64];
@@ -73,6 +96,7 @@
 int blocked_counter = 0;
 int file_vfd = -1;
 bool active_icon[16] = { false };
+int adb_led = -1;
 
 pthread_t vfd_scrollText;
 
@@ -82,6 +106,33 @@
 	unsigned char data[64];
 	unsigned char length;
 };
+
+#if defined (BOXMODEL_ADB_BOX)
+int get_adb_variant()
+{
+	FILE *adb_file;
+	char adb_model[5] = { 0x00 };
+	int  ret;
+
+	adb_file = fopen("/proc/stb/info/adb_variant", "r");
+	if (adb_file != NULL)
+	{
+		fread(adb_model, 4, 1, adb_file);
+		printf("[CVFD] get_adb_variant: detected ADB model %s\n", adb_model);
+		if (strcmp(adb_model, "bska") == 0 || strcmp(adb_model, "bxzb") == 0)
+		{
+			printf("[CVFD] get_adb_variant: LED display assumed.\n");
+			ret = 1;
+		}
+		else
+		{
+			printf("[CVFD] get_adb_variant: VFD display assumed.\n");
+			ret = 0;
+		}
+	}
+	return ret;
+}
+#endif
 
 static void write_to_vfd(unsigned int DevType, struct vfd_ioctl_data *data, bool force = false)
 {
@@ -147,6 +198,17 @@
 
 static void ShowNormalText(char *str, bool fromScrollThread = false)
 {
+	int colon_at_2 = 0;
+	unsigned int disp_len = 0;
+
+#if defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7819)
+	if (str[2] == 0x3a)
+	{
+		colon_at_2++;
+	}
+#endif
 	if (blocked)
 	{
 		printf("[CVFD] - blocked\n");
@@ -166,32 +228,69 @@
 			vfd_scrollText = 0;
 		}
 	}
-	if ((strlen(str) > VFDLENGTH && !fromScrollThread) && (g_settings.lcd_vfd_scroll >= 1))
+	disp_len = (adb_led == 1 ? 4 : VFDLENGTH);
+#if defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7819)
+	if ((strlen(str) > disp_len + colon_at_2 && !fromScrollThread) && (g_settings.lcd_vfd_scroll >= 1))
+#else
+	if ((strlen(str) > disp_len && !fromScrollThread) && (g_settings.lcd_vfd_scroll >= 1))
+#endif
 	{
 		CVFD::getInstance()->ShowScrollText(str);
 		return;
 	}
-
-	if (strlen(str) < VFDLENGTH && VFDLENGTH > 7) // do not center on small displays
-		ws = (VFDLENGTH - strlen(str)) / 2;
+#if defined (BOXMODEL_CUBEREVO_MINI) \
+ || defined (BOXMODEL_CUBEREVO_MINI2) \
+ || defined (BOXMODEL_CUBEREVO_MINI_FTA) \
+ || defined (BOXMODEL_CUBEREVO_250HD) \
+ || defined (BOXMODEL_CUBEREVO_2000HD) \
+ || defined (BOXMODEL_CUBEREVO_3000HD) \
+ || defined (BOXMODEL_CUBEREVO_9500HD) \
+ || defined (BOXMODEL_CUBEREVO)  // CubeRevo's center in the display driver
+	ws = 0;
+#else
+	if (strlen(str) < disp_len + colon_at_2 && disp_len > 7) // do not center on small displays
+	{
+		ws = (disp_len - strlen(str)) / 2;
+	}
 	else
+	{
 		ws = 0;
+	}
+#endif
 	memset(data.data, ' ', 63);
 	if (!fromScrollThread)
 	{
-		memcpy(data.data + ws, str, VFDLENGTH - ws);
-		data.start = 0;
+#if defined (BOXMODEL_CUBEREVO_MINI) \
+ || defined (BOXMODEL_CUBEREVO_MINI2) \
+ || defined (BOXMODEL_CUBEREVO_MINI_FTA) \
+ || defined (BOXMODEL_CUBEREVO_250HD) \
+ || defined (BOXMODEL_CUBEREVO_2000HD) \
+ || defined (BOXMODEL_CUBEREVO_3000HD) \
+ || defined (BOXMODEL_CUBEREVO_9500HD) \
+ || defined (BOXMODEL_CUBEREVO)  // CubeRevo's center in the display driver
+		memcpy(data.data, str, strlen(str));
+		data.length = strlen(str);
+#else
+		memcpy(data.data + ws, str, disp_len - ws + colon_at_2);
 		if ((strlen(str) % 2) == 1 && VFDLENGTH > 7) // do not center on small displays
-			data.length = VFDLENGTH - ws - 1;
+		{
+			data.length = disp_len - ws - 1 + colon_at_2;
+		}
 		else
-			data.length = VFDLENGTH - ws;
+		{
+			data.length = disp_len - ws + colon_at_2;
+		}
+#endif
 	}
 	else
 	{
-		memcpy(data.data, str, VFDLENGTH);
-		data.start = 0;
-		data.length = VFDLENGTH;
-	}
+		memcpy(data.data, str, disp_len + colon_at_2);
+		data.length = disp_len + colon_at_2;
+	}
+	data.start = 0;
+//	printf("CVFD::ShowNormalText: [%s]\n", str);
 	write_to_vfd(VFDDISPLAYCHARS, &data);
 	return;
 }
@@ -227,17 +326,18 @@
 	int i;
 	char *str = (char *)arg;
 	int len = strlen(str);
-	char out[VFDLENGTH + 1];
-	char buf[VFDLENGTH + 65];
-
-	memset(out, 0, VFDLENGTH + 1);
+	char out[17];
+	char buf[81];
+	int disp_len = adb_led == 1 ? 4 : VFDLENGTH;
+
+	memset(out, 0, disp_len + 1);
 
 	int retries = g_settings.lcd_vfd_scroll;
 
-	if (len > VFDLENGTH)
+	if (len > disp_len)
 	{
 		printf("CVFD::ThreadScrollText: [%s], length %d\n", str, len);
-		memset(buf, ' ', (len + VFDLENGTH));
+		memset(buf, ' ', (len + disp_len));
 		memcpy(buf, str, len);
 
 		while (retries--)
@@ -247,20 +347,20 @@
 			for (i = 0; i <= (len - 1); i++)
 			{
 				// scroll text until end
-				memcpy(out, buf + i, VFDLENGTH);
+				memcpy(out, buf + i, disp_len);
 				ShowNormalText(out, true);
 				usleep(SCROLL_TIME);
 			}
 		}
 	}
-	memcpy(out, str, VFDLENGTH); // display first VFDLENGTH chars after scrolling
+	memcpy(out, str, disp_len); // display first VFDLENGTH chars after scrolling
 	ShowNormalText(out, true);
 
 	pthread_exit(0);
 
 	return NULL;
 }
-#endif
+#endif // HAVE_DUCKBOX_HARDWARE
 
 CVFD::CVFD()
 {
@@ -373,7 +473,7 @@
 	{
 		switch_name_time_cnt = g_settings.timing[SNeutrinoSettings::TIMING_INFOBAR] + 10;
 	}
-#if defined (BOXMODEL_OCTAGON1008)
+#if defined (BOXMODEL_HS9510)
 	ShowIcon(ICON_COLON2, false);
 #endif
 
@@ -398,6 +498,9 @@
 
 void CVFD::init(const char * /*fontfile*/, const char * /*fontname*/)
 {
+#if defined (BOXMODEL_ADB_BOX)
+	adb_led = get_adb_variant();
+#endif		
 	//InitNewClock(); /FIXME
 
 	brightness = -1;
@@ -412,68 +515,139 @@
 
 void CVFD::setlcdparameter(int dimm, const int power)
 {
-	if (fd < 0) return;
-
-	if (dimm < 0)
+	if (fd < 0)
+	{
+		return;
+	}
+#if defined (BOXMODEL_HS9510) \
+ || defined (BOXMODEL_FS9000) \
+ || defined (BOXMODEL_HS8200) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_CUBEREVO) \
+ || defined (BOXMODEL_CUBEREVO_MINI) \
+ || defined (BOXMODEL_CUBEREVO_MINI2) \
+ || defined (BOXMODEL_CUBEREVO_2000HD) \
+ || defined (BOXMODEL_CUBEREVO_3000HD) \
+ || defined (BOXMODEL_IPBOX9900) \
+ || defined (BOXMODEL_IPBOX99) \
+ || defined (BOXMODEL_ADB_BOX) \
+ || defined (BOXMODEL_UFS912) \
+ || defined (BOXMODEL_UFS913) \
+ || defined (BOXMODEL_UFS922)
+	else if(dimm > 7)
+	{
+		dimm = 7;
+	}
+#else
+	else if (dimm > 15)
+	{
+		dimm = 15;
+	}
+#endif
+	if (!power)
+	{
 		dimm = 0;
-	else if (dimm > 15)
-		dimm = 15;
-
-	if (!power)
-		dimm = 0;
-
+	}
 	if (brightness == dimm)
-		return;
-
+	{
+		return;
+	}
 	brightness = dimm;
 
 	printf("CVFD::setlcdparameter dimm %d power %d\n", dimm, power);
 #if !HAVE_DUCKBOX_HARDWARE
 	int ret = ioctl(fd, IOC_FP_SET_BRIGHT, dimm);
 	if (ret < 0)
+	{
 		perror("IOC_FP_SET_BRIGHT");
+	}
 #else
 // Brightness
 	struct vfd_ioctl_data data;
+
 	memset(&data, 0, sizeof(struct vfd_ioctl_data));
 	data.start = brightness & 0x07;
 	data.length = 0;
 	write_to_vfd(VFDBRIGHTNESS, &data);
-#endif
-#if defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#if defined (BOXMODEL_FS9000) \
+ || defined (BOXMODEL_HS8200)
 	usleep(100000);
 	memset(&data, 0, sizeof(struct vfd_ioctl_data));
-	data.start = 0;
+	data.data[0] = 0;
 	data.length = 5;
 	if (power)
 	{
-		data.data[0] = 0x01; // red led
+		data.start = 0x01;  // red led
 	}
 	else
 	{
-		data.data[0] = 0xf2; // cross plus blue led
-	}
-	data.start = 0;
-	data.data[4] = 0; // off
+		data.start = 0xf2;  // cross plus blue led
+	}
+	data.data[4] = 0;  // off
 	data.length = 5;
-	write_to_vfd(VFDPWRLED, &data);
+	write_to_vfd(VFDSETLED, &data);
 	usleep(100000);
 	memset(&data, 0, sizeof(struct vfd_ioctl_data));
-	data.start = 0;
+	data.data[0] = 0;
 	data.length = 5;
 	if (power)
 	{
-		data.data[0] = 0xf2; // cross plus blue led
+		data.start = 0xf2;  // cross plus blue led
 	}
 	else
 	{
-		data.data[0] = 0x01; // red led
+		data.start = 0x01;  // red led
 	}
 	data.start = 0;
 	data.data[4] = brightness * 2;
 	data.length = 5;
-	write_to_vfd(VFDPWRLED, &data);
-#elif !defined (BOXMODEL_UFS912) && !defined (BOXMODEL_UFS913) && !defined (BOXMODEL_OCTAGON1008)
+	write_to_vfd(VFDSETLED, &data);
+#elif defined (BOXMODEL_HS7420) \
+ ||   defined (BOXMODEL_HS7810A) \
+ ||   defined (BOXMODEL_HS7429) \
+ ||   defined (BOXMODEL_HS7819)
+// TODO: FIX all models
+	memset(&data, 0, sizeof(struct vfd_ioctl_data));
+	data.start = 0;
+	data.data[0] = 0x02; // logo
+	data.data[4] = (brightness & 0x07);
+	data.length = 5;
+	write_to_vfd(VFDSETLED, &data);
+// TODO: Octagon1008
+#elif defined (BOXMODEL_ADB_BOX)
+	memset(&data, 0, sizeof(struct vfd_ioctl_data));
+	data.start = brightness & 0x07;
+	data.length = 0;
+	if (adb_led == 1)
+	{
+		write_to_vfd(VFDLEDBRIGHTNESS, &data);
+	}
+	else
+	{
+		write_to_vfd(VFDBRIGHTNESS, &data);
+	}
+//	memset(&data, 0, sizeof(struct vfd_ioctl_data));
+//	data.start = 0x01; // power LED
+//	data.data[0] = 0;
+//	data.data[1] = 0;
+//	data.data[4] = 0;
+//	data.length = 5;
+//	if (power)
+//	{
+//		data.data[3] = 1;  // green
+//	}
+//	else
+//	{
+//		data.data[3] = 2;  // red
+//	}
+//	write_to_vfd(VFDSETLED, &data);
+#elif !defined (BOXMODEL_UFS912) \
+ &&   !defined (BOXMODEL_UFS913) \
+ &&   !defined (BOXMODEL_UFS922) \
+ &&   !defined (BOXMODEL_HS9510) \
+ &&   !defined (BOXMODEL_OPT9600) \
+ &&   !defined (BOXMODEL_OPT9600MINI)
 // Power on/off
 	if (power)
 	{
@@ -486,11 +660,15 @@
 	data.length = 0;
 	write_to_vfd(VFDDISPLAYWRITEONOFF, &data, true);
 #endif
+#endif
 }
 
 void CVFD::setlcdparameter(void)
 {
-	if (fd < 0) return;
+	if (fd < 0)
+	{
+		return;
+	}
 	last_toggle_state_power = g_settings.lcd_setting[SNeutrinoSettings::LCD_POWER];
 	setlcdparameter((mode == MODE_STANDBY) ? g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS] : (mode == MODE_SHUTDOWN) ? g_settings.lcd_setting[SNeutrinoSettings::LCD_DEEPSTANDBY_BRIGHTNESS] : g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS],
 		last_toggle_state_power);
@@ -498,19 +676,32 @@
 
 void CVFD::showServicename(const std::string &name, int number)  // UTF-8
 {
-	if (fd < 0) return;
-
-	printf("CVFD::showServicename: %s\n", name.c_str());
+	int disp_len = (adb_led == 1 ? 4 : VFDLENGTH);
+
+	if (fd < 0)
+	{
+		return;
+	}
+//	printf("CVFD::showServicename: [%s] (len = %d\n", name.c_str(), strlen(name.c_str()));
 	servicename = name;
 	service_number = number;
 
 	if (mode != MODE_TVRADIO)
-		return;
-
-	if (support_text)
+	{
+		return;
+	}
+	if (strlen(name.c_str()) < 1)
+	{
+		return;
+	}
+	if ((support_text) && disp_len > 7)
+	{
 		ShowText(name.c_str());
+	}
 	else
+	{
 		ShowNumber(service_number);
+	}
 	wake_up();
 }
 
@@ -518,9 +709,12 @@
 {
 	//unsigned int system_rev = cs_get_revision();
 	static int recstatus = 0;
+
 #if 0
 	if (!has_lcd)
-		return;
+	{
+		return;
+	}
 #endif
 	if (fd >= 0 && mode == MODE_SHUTDOWN)
 	{
@@ -534,18 +728,53 @@
 			char timestr[21];
 			struct timeb tm;
 			struct tm *t;
-			static int hour = 0, minute = 0;
+			static int hour = 0, minute = 0, second = 0;
 
 			ftime(&tm);
 			t = localtime(&tm.time);
-			if (force || (switch_name_time_cnt == 0 && ((hour != t->tm_hour) || (minute != t->tm_min))))
+			if (force || (switch_name_time_cnt == 0 && ((hour != t->tm_hour) || (minute != t->tm_min) || (second != t->tm_sec))))
 			{
 				hour = t->tm_hour;
 				minute = t->tm_min;
-#if defined (BOXMODEL_OCTAGON1008)
-				ShowIcon(ICON_COLON2, true);
-#elif defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_CUBEREVO_250HD)
+				second = t->tm_sec;
+#if defined (BOXMODEL_HS9510)
+				ShowIcon(ICON_COLON2, second % 2 == 0 ? true : false);
+ 				strftime(timestr, 5, "%H%M", t);
+				ShowText(timestr);
+#elif defined (BOXMODEL_HS7420) \
+ ||   defined (BOXMODEL_HS7429)
+				ShowIcon(ICON_COLON2, second % 2 == 0 ? true : false);
 				strftime(timestr, 5, "%H%M", t);
+				ShowText(timestr);
+#elif defined (BOXMODEL_HS7119) \
+ ||   defined (BOXMODEL_HS7810A) \
+ ||   defined (BOXMODEL_HS7819)
+				strftime(timestr, 5, "%H%M", t);
+				ShowText(timestr);
+				ShowIcon(ICON_COLON, second % 2 == 0 ? true : false);
+#elif defined (BOXMODEL_FS9000)
+				ShowIcon(ICON_COLON3, second % 2 == 0 ? true : false);
+				strftime(timestr, 13, "        %H%M", t);
+				ShowText(timestr);
+#elif defined (BOXMODEL_ADB_BOX)
+				if (adb_led == 1)
+				{
+					strftime(timestr, 5, "%H%M", t);
+				}
+				else
+				{
+					strftime(timestr, 6, (second % 2 == 0 ? "%H:%M" : "%H %M"), t);
+				}
+				ShowText(timestr);
+#elif defined (BOXMODEL_OPT9600)
+				strftime(timestr, 6, (second % 2 == 0 ? "%H-%M" : "%H %M"), t);
+				ShowText(timestr);
+#elif defined (BOXMODEL_CUBEREVO_250HD) \
+ ||   defined (BOXMODEL_CUBEREVO_MINI_FTA) \
+ ||   defined (BOXMODEL_ATEMIO520) \
+ ||   defined (BOXMODEL_OPT9600MINI)
+				strftime(timestr, 5, "%H%M", t);
+				ShowText(timestr);
 #else
 				strftime(timestr, 6, "%H:%M", t);
 #endif
@@ -570,20 +799,26 @@
 		{
 			clearClock = 0;
 			if (has_lcd)
+			{
 				ShowIcon(FP_ICON_CAM1, false);
+			}
 		}
 		else
 		{
 			clearClock = 1;
 			if (has_lcd)
+			{
 				ShowIcon(FP_ICON_CAM1, true);
-		}
-	}
-	else if (clearClock || (recstatus != tmp_recstatus))    // in case icon ON after record stopped
+			}
+		}
+	}
+	else if (clearClock || (recstatus != tmp_recstatus))  // in case icon ON after record stopped
 	{
 		clearClock = 0;
 		if (has_lcd)
+		{
 			ShowIcon(FP_ICON_CAM1, false);
+		}
 	}
 
 	recstatus = tmp_recstatus;
@@ -598,7 +833,9 @@
 		ShowIcon(FP_ICON_HD, chan->isHD());
 		ShowIcon(FP_ICON_LOCK, !chan->camap.empty());
 		if (chan->getAudioChannel() != NULL)
+		{
 			ShowIcon(FP_ICON_DD, chan->getAudioChannel()->audioChannelType == CZapitAudioChannel::AC3);
+		}
 	}
 }
 #endif
@@ -620,18 +857,23 @@
 void CVFD::showVolume(const char vol, const bool force_update)
 {
 	static int oldpp = 0;
-	if (!has_lcd) return;
-
+	if (!has_lcd)
+	{
+		return;
+	}
 	ShowIcon(FP_ICON_MUTE, muted);
 
 	if (!force_update && vol == volume)
-		return;
+	{
+		return;
+	}
 	volume = vol;
 
 	bool allowed_mode = (mode == MODE_TVRADIO || mode == MODE_AUDIO || mode == MODE_MENU_UTF8);
 	if (!allowed_mode)
-		return;
-
+	{
+		return;
+	}
 	if (g_settings.lcd_setting[SNeutrinoSettings::LCD_SHOW_VOLUME] == 1)
 	{
 		wake_up();
@@ -639,7 +881,8 @@
 		int pp = (int) round((double) vol / (double) 2);
 		if (oldpp != pp)
 		{
-#if defined (BOXMODEL_UFS910) || defined (BOXMODEL_UFS922)
+#if defined (BOXMODEL_UFS910) \
+ || defined (BOXMODEL_UFS922)
 			int i;
 			unsigned char speaker[5] = {0x1C, 0x1C, 0x1C, 0x3E, 0x7F}; // speaker symbol
 			writeCG(0, speaker);
@@ -680,22 +923,60 @@
 			if (strlen(VolumeBar) < 12)
 			{
 				for (int a = strlen(VolumeBar); a < 12; a++)
+				{
 					strncat(VolumeBar, c0, 1);
+				}
 			}
 			ShowText(VolumeBar);
 #elif defined (BOXMODEL_TF7700)
 			char vol_chr[64] = "";
 			snprintf(vol_chr, sizeof(vol_chr) - 1, "VOL: %d%%", (int)vol);
 			ShowText(vol_chr);
-#elif defined (BOXMODEL_OCTAGON1008)
+#elif defined (BOXMODEL_OPT9600)
 			char vol_chr[64] = "";
-			snprintf(vol_chr, sizeof(vol_chr) - 1, "VOL=%3d", (int)vol);
+			snprintf(vol_chr, sizeof(vol_chr) - 1, "VOL %3d%%", (int)vol);
 			ShowText(vol_chr);
-#elif defined (BOXMODEL_CUBEREVO_250HD) || defined (BOXMODEL_IPBOX55)
+#elif defined (BOXMODEL_HS9510) \
+ ||   defined (BOXMODEL_HS7420) \
+ ||   defined (BOXMODEL_HS7429)
 			char vol_chr[64] = "";
+			snprintf(vol_chr, sizeof(vol_chr) - 1, "VOL=%3d%%", (int)vol);
+			ShowText(vol_chr);
+#elif defined (BOXMODEL_HS7119) \
+ ||   defined (BOXMODEL_HS7810A) \
+ ||   defined (BOXMODEL_HS7819) \
+ ||   defined (BOXMODEL_CUBEREVO_MINI_FTA) \
+ ||   defined (BOXMODEL_CUBEREVO_250HD) \
+ ||   defined (BOXMODEL_IPBOX55) \
+ ||   defined (BOXMODEL_ATEMIO520) \
+ ||   defined (BOXMODEL_OPT9600MINI)
+			char vol_chr[64] = "";
+
 			snprintf(vol_chr, sizeof(vol_chr) - 1, "v%3d", (int)vol);
 			ShowText(vol_chr);
-#elif defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500) || defined (BOXMODEL_UFS912) || defined (BOXMODEL_UFS913) || defined (BOXMODEL_CUBEREVO) || defined (BOXMODEL_CUBEREVO_MINI) || defined (BOXMODEL_CUBEREVO_MINI2) || defined (BOXMODEL_CUBEREVO_2000HD) || defined (BOXMODEL_CUBEREVO_3000HD) || defined (BOXMODEL_IPBOX9900) || defined (BOXMODEL_IPBOX99)
+#elif defined (BOXMODEL_ADB_BOX)
+			char vol_chr[64] = "";
+
+			if (adb_led == 1)
+			{
+				snprintf(vol_chr, sizeof(vol_chr) - 1, "v%3d", (int)vol);
+			}
+			else
+			{
+				snprintf(vol_chr, sizeof(vol_chr) - 1, "Volume: %d%%", (int)vol);
+			}
+			ShowText(vol_chr);
+#elif defined (BOXMODEL_FS9000) \
+ ||   defined (BOXMODEL_HS8200) \
+ ||   defined (BOXMODEL_UFS912) \
+ ||   defined (BOXMODEL_UFS913) \
+ ||   defined (BOXMODEL_CUBEREVO) \
+ ||   defined (BOXMODEL_CUBEREVO_MINI) \
+ ||   defined (BOXMODEL_CUBEREVO_MINI2) \
+ ||   defined (BOXMODEL_CUBEREVO_2000HD) \
+ ||   defined (BOXMODEL_CUBEREVO_3000HD) \
+ ||   defined (BOXMODEL_IPBOX9900) \
+ ||   defined (BOXMODEL_IPBOX99)
 			char vol_chr[64] = "";
 			snprintf(vol_chr, sizeof(vol_chr) - 1, "Volume: %d%%", (int)vol);
 			ShowText(vol_chr);
@@ -705,8 +986,10 @@
 #else
 		ShowIcon(FP_ICON_FRAME, true);
 		int pp = (int) round((double) vol * (double) 8 / (double) 100);
-		if (pp > 8) pp = 8;
-
+		if (pp > 8)
+		{
+			pp = 8;
+		}
 		if (force_update || oldpp != pp)
 		{
 			printf("CVFD::showVolume: %d, bar %d\n", (int) vol, pp);
@@ -735,30 +1018,40 @@
 #else
 
 	static int ppold = 0;
-	if (!has_lcd) return;
-
+	if (!has_lcd)
+	{
+		return;
+	}
 	percentOver = perc;
 
 	if (mode == MODE_AUDIO && origin != MODE_AUDIO) // exclusive access for audio mode
-		return;
-
+	{
+		return;
+	}
 	bool allowed_mode = (mode == MODE_TVRADIO || mode == MODE_AUDIO || mode == MODE_MENU_UTF8);
 	if (!allowed_mode)
-		return;
-
+	{
+		return;
+	}
 	if (g_settings.lcd_setting[SNeutrinoSettings::LCD_SHOW_VOLUME] == 0)
 	{
 		ShowIcon(FP_ICON_FRAME, true);
 		int pp;
 		if (perc == 255)
+		{
 			pp = 0;
+		}
 		else
+		
 			pp = (int) round((double) perc * (double) 8 / (double) 100);
-		if (pp > 8) pp = 8;
-
+		}
+		if (pp > 8)
+		{
+			pp = 8;
+		}
 		if (pp != ppold)
 		{
-//printf("CVFD::showPercentOver: %d, bar %d\n", (int) perc, pp);
+//			printf("CVFD::showPercentOver: %d, bar %d\n", (int) perc, pp);
 			int i;
 			int j = 0x00000200;
 			for (i = 0; i < pp; i++)
@@ -779,19 +1072,28 @@
 
 void CVFD::showMenuText(const int /*position*/, const char *ptext, const int /*highlight*/, const bool /*utf_encoded*/)
 {
-	if (fd < 0) return;
+	if (fd < 0)
+	{
+		return;
+	}
 	if (mode != MODE_MENU_UTF8)
-		return;
-
+	{
+		return;
+	}
 	ShowText(ptext);
 	wake_up();
 }
 
 void CVFD::showAudioTrack(const std::string & /*artist*/, const std::string &title, const std::string & /*album*/)
 {
-	if (fd < 0) return;
+	if (fd < 0)
+	{
+		return;
+	}
 	if (mode != MODE_AUDIO)
-		return;
+	{
+		return;
+	}
 	printf("CVFD::showAudioTrack: %s\n", title.c_str());
 	ShowText(title.c_str());
 	wake_up();
@@ -805,7 +1107,10 @@
 
 void CVFD::showAudioPlayMode(AUDIOMODES m)
 {
-	if (fd < 0) return;
+	if (fd < 0)
+	{
+		return;
+	}
 	if (mode != MODE_AUDIO)
 		return;
 
@@ -853,21 +1158,33 @@
 
 void CVFD::showAudioProgress(const unsigned char perc)
 {
-	if (fd < 0) return;
+	if (fd < 0)
+	{
+		return;
+	}
 	if (mode != MODE_AUDIO)
-		return;
-
+	{
+		return;
+	}
 	showPercentOver(perc, true, MODE_AUDIO);
 }
 
 void CVFD::setMode(const MODES m, const char *const title)
 {
-	if (fd < 0) return;
-
+#if defined (BOXMODEL_ADB_BOX)
+	struct vfd_ioctl_data data;
+
+#endif
+	if (fd < 0)
+	{
+		return;
+	}
 	// Clear colon in display if it is still there
 
 	if (mode == MODE_AUDIO)
+	{
 		ShowIcon(FP_ICON_MP3, false);
+	}
 #if 0
 	else if (mode == MODE_STANDBY)
 	{
@@ -876,8 +1193,43 @@
 	}
 #endif
 
+#if defined (BOXMODEL_HS9510) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429)
+	else if (mode == MODE_STANDBY)
+	{
+		ShowIcon(ICON_COLON2, true);
+	}
+#elif defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7819)
+	else if (mode == MODE_STANDBY)
+	{
+		ShowIcon(ICON_COLON, true);
+	}
+#elif defined (BOXMODEL_FS9000)
+	else if (mode == MODE_STANDBY)
+	{
+		ShowIcon(ICON_COLON3, true);
+	}
+#elif defined (BOXMODEL_ADB_BOX)
+	else if (mode == MODE_STANDBY)
+	{
+		memset(&data, 0, sizeof(struct vfd_ioctl_data));
+		data.start = 0x01; // power LED
+		data.data[0] = 0;
+		data.data[1] = 0;
+		data.data[3] = 2;  // red
+		data.data[4] = 0;
+		data.length = 5;
+		write_to_vfd(VFDSETLED, &data);
+	}
+#endif
+
 	if (strlen(title))
+	{
 		ShowText(title);
+	}
 	mode = m;
 	setlcdparameter();
 
@@ -902,7 +1254,9 @@
 			showServicename(servicename);
 			showclock = true;
 			if (g_settings.lcd_info_line)
+			{
 				switch_name_time_cnt = g_settings.timing[SNeutrinoSettings::TIMING_INFOBAR] + 10;
+			}
 			break;
 		case MODE_AUDIO:
 		{
@@ -965,8 +1319,10 @@
 
 void CVFD::setBrightness(int bright)
 {
-	if (!has_lcd && !has_led_segment) return;
-
+	if (!has_lcd && !has_led_segment)
+	{
+		return;
+	}
 	g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS] = bright;
 	setlcdparameter();
 }
@@ -974,21 +1330,41 @@
 int CVFD::getBrightness()
 {
 	//FIXME for old neutrino.conf
-#if defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#if defined (BOXMODEL_HS9510) \
+ || defined (BOXMODEL_FS9000) \
+ || defined (BOXMODEL_HS8200) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_CUBEREVO) \
+ || defined (BOXMODEL_CUBEREVO_MINI) \
+ || defined (BOXMODEL_CUBEREVO_MINI2) \
+ || defined (BOXMODEL_CUBEREVO_2000HD) \
+ || defined (BOXMODEL_CUBEREVO_3000HD) \
+ || defined (BOXMODEL_IPBOX9900) \
+ || defined (BOXMODEL_IPBOX99) \
+ || defined (BOXMODEL_ADB_BOX) \
+ || defined (BOXMODEL_UFS912) \
+ || defined (BOXMODEL_UFS913) \
+ || defined (BOXMODEL_UFS922)
 	if (g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS] > 7)
+	{
 		g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS] = 7;
+	}
 #else
 	if (g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS] > 15)
+	{
 		g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS] = 15;
-#endif
-
+	}
+#endif
 	return g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS];
 }
 
 void CVFD::setBrightnessStandby(int bright)
 {
-	if (!has_lcd && !has_led_segment) return;
-
+	if (!has_lcd && !has_led_segment)
+	{
+		return;
+	}
 	g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS] = bright;
 	setlcdparameter();
 }
@@ -996,20 +1372,41 @@
 int CVFD::getBrightnessStandby()
 {
 	//FIXME for old neutrino.conf
-#if defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#if defined (BOXMODEL_HS9510) \
+ || defined (BOXMODEL_FS9000) \
+ || defined (BOXMODEL_HS8200) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_CUBEREVO) \
+ || defined (BOXMODEL_CUBEREVO_MINI) \
+ || defined (BOXMODEL_CUBEREVO_MINI2) \
+ || defined (BOXMODEL_CUBEREVO_2000HD) \
+ || defined (BOXMODEL_CUBEREVO_3000HD) \
+ || defined (BOXMODEL_IPBOX9900) \
+ || defined (BOXMODEL_IPBOX99) \
+ || defined (BOXMODEL_ADB_BOX) \
+ || defined (BOXMODEL_UFS912) \
+ || defined (BOXMODEL_UFS913) \
+ || defined (BOXMODEL_UFS922)
 	if (g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS] > 7)
+	{
 		g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS] = 7;
+	}
 #else
 	if (g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS] > 15)
+	{
 		g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS] = 15;
+	}
 #endif
 	return g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS];
 }
 
 void CVFD::setBrightnessDeepStandby(int bright)
 {
-	if (!has_lcd && !has_led_segment) return;
-
+	if (!has_lcd && !has_led_segment)
+	{
+		return;
+	}
 	g_settings.lcd_setting[SNeutrinoSettings::LCD_DEEPSTANDBY_BRIGHTNESS] = bright;
 	setlcdparameter();
 }
@@ -1017,19 +1414,38 @@
 int CVFD::getBrightnessDeepStandby()
 {
 	//FIXME for old neutrino.conf
-#if defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#if defined (BOXMODEL_HS9510) \
+ || defined (BOXMODEL_FS9000) \
+ || defined (BOXMODEL_HS8200) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_CUBEREVO) \
+ || defined (BOXMODEL_CUBEREVO_MINI) \
+ || defined (BOXMODEL_CUBEREVO_MINI2) \
+ || defined (BOXMODEL_CUBEREVO_2000HD) \
+ || defined (BOXMODEL_CUBEREVO_3000HD) \
+ || defined (BOXMODEL_IPBOX9900) \
+ || defined (BOXMODEL_IPBOX99) \
+ || defined (BOXMODEL_ADB_BOX)
 	if (g_settings.lcd_setting[SNeutrinoSettings::LCD_DEEPSTANDBY_BRIGHTNESS] > 7)
+	{
 		g_settings.lcd_setting[SNeutrinoSettings::LCD_DEEPSTANDBY_BRIGHTNESS] = 7;
+	}
 #else
 	if (g_settings.lcd_setting[SNeutrinoSettings::LCD_DEEPSTANDBY_BRIGHTNESS] > 15)
+	{
 		g_settings.lcd_setting[SNeutrinoSettings::LCD_DEEPSTANDBY_BRIGHTNESS] = 15;
+	}
 #endif
 	return g_settings.lcd_setting[SNeutrinoSettings::LCD_DEEPSTANDBY_BRIGHTNESS];
 }
 
 void CVFD::setPower(int /*power*/)
 {
-	if (!has_lcd) return;
+	if (!has_lcd)
+	{
+		return;
+	}
 // old
 	//g_settings.lcd_setting[SNeutrinoSettings::LCD_POWER] = power;
 	//setlcdparameter();
@@ -1042,8 +1458,10 @@
 
 void CVFD::togglePower(void)
 {
-	if (fd < 0) return;
-
+	if (fd < 0)
+	{
+		return;
+	}
 	last_toggle_state_power = 1 - last_toggle_state_power;
 	setlcdparameter((mode == MODE_STANDBY) ? g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS] : (mode == MODE_SHUTDOWN) ? g_settings.lcd_setting[SNeutrinoSettings::LCD_DEEPSTANDBY_BRIGHTNESS] : g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS],
 		last_toggle_state_power);
@@ -1051,51 +1469,95 @@
 
 void CVFD::setMuted(bool mu)
 {
-	if (!has_lcd) return;
+	if (!has_lcd)
+	{
+		return;
+	}
 	muted = mu;
 	showVolume(volume);
 }
 
 void CVFD::resume()
 {
-	if (!has_lcd) return;
+	if (!has_lcd)
+	{
+		return;
+	}
 }
 
 void CVFD::pause()
 {
-	if (!has_lcd) return;
+	if (!has_lcd)
+	{
+		return;
+	}
 }
 
 void CVFD::Lock()
 {
-	if (fd < 0) return;
+	if (fd < 0)
+	{
+		return;
+	}
 	creat("/tmp/vfd.locked", 0);
 }
 
 void CVFD::Unlock()
 {
-	if (fd < 0) return;
+	if (fd < 0)
+	{
+		return;
+	}
 	unlink("/tmp/vfd.locked");
 }
 
 void CVFD::Clear()
 {
-	if (fd < 0) return;
+	if (fd < 0)
+	{
+		return;
+	}
 #if !HAVE_DUCKBOX_HARDWARE
 	int ret = ioctl(fd, IOC_FP_CLEAR_ALL, 0);
 	if (ret < 0)
+	{
 		perror("IOC_FP_SET_TEXT");
+	}
 	else
+	{
 		text.clear();
+	}
 #else
-#if defined (BOXMODEL_CUBEREVO_250HD) || defined (BOXMODEL_IPBOX55)
+#if defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7819) \
+ || defined (BOXMODEL_CUBEREVO_MINI_FTA) \
+ || defined (BOXMODEL_CUBEREVO_250HD) \
+ || defined (BOXMODEL_IPBOX55) \
+ || defined (BOXMODEL_ATEMIO520) \
+ || defined (BOXMODEL_OPT9600MINI)
 	ShowText("    ");
-#elif defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_TF7700)
+#elif defined (BOXMODEL_HS9510) \
+ ||   defined (BOXMODEL_TF7700) \
+ ||   defined (BOXMODEL_HS7420) \
+ ||   defined (BOXMODEL_HS7429) \
+ ||   defined (BOXMODEL_OPT9600)
 	ShowText("        ");
-#elif defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#elif defined (BOXMODEL_FS9000) \
+ ||   defined (BOXMODEL_HS8200)
 	ShowText("            ");
-#elif defined (BOXMODEL_IPBOX9900) || defined (BOXMODEL_IPBOX99)
+#elif defined (BOXMODEL_CUBEREVO) \
+ ||   defined (BOXMODEL_CUBEREVO_MINI) \
+ ||   defined (BOXMODEL_CUBEREVO_MINI2) \
+ ||   defined (BOXMODEL_CUBEREVO_2000HD) \
+ ||   defined (BOXMODEL_CUBEREVO_3000HD) \
+ ||   defined (BOXMODEL_IPBOX9900) \
+ ||   defined (BOXMODEL_IPBOX99)
 	ShowText("              ");
+#elif defined (BOXMODEL_ADB_BOX)
+	ShowText("               ");
+#elif !defined (BOXMODEL_HS7110)
+	ShowText("                ");
 #endif
 #endif
 }
@@ -1103,18 +1565,36 @@
 void CVFD::ShowIcon(fp_icon icon, bool show)
 {
 #if !HAVE_DUCKBOX_HARDWARE
-	if (!has_lcd || fd < 0) return;
-//printf("CVFD::ShowIcon %s %x\n", show ? "show" : "hide", (int) icon);
+	if (!has_lcd || fd < 0)
+	{
+		return;
+	}
+//	printf("CVFD::ShowIcon %s %x\n", show ? "show" : "hide", (int) icon);
+
 	int ret = ioctl(fd, show ? IOC_FP_SET_ICON : IOC_FP_CLEAR_ICON, icon);
 	if (ret < 0)
+	{
 		perror(show ? "IOC_FP_SET_ICON" : "IOC_FP_CLEAR_ICON");
+	}
 #else
-#if defined (BOXMODEL_ATEVIO7500)
+  #if defined (BOXMODEL_HS7110) \
+   || defined (BOXMODEL_HS7420) \
+   || defined (BOXMODEL_HS7810A) \
+   || defined (BOXMODEL_HS7119) \
+   || defined (BOXMODEL_HS7429) \
+   || defined (BOXMODEL_HS7819) \
+   || defined (BOXMODEL_CUBEREVO_MINI_FTA) \
+   || defined (BOXMODEL_CUBEREVO_250HD) \
+   || defined (BOXMODEL_IPBOX55) \
+   || defined (BOXMODEL_ATEMIO520) \
+   || defined (BOXMODEL_OPT9600) \
+   || defined (BOXMODEL_OPT9600MINI)
 	return;
-#endif
+  #endif
 	if (icon == 0)
-		return;
-
+	{
+		return;
+	}
 	if (blocked)
 	{
 		printf("[CVFD] - blocked\n");
@@ -1122,10 +1602,13 @@
 	}
 
 	if (active_icon[icon & 0x0F] == show)
-		return;
+	{
+		return;
+	}
 	else
+	{
 		active_icon[icon & 0x0F] = show;
-
+	}
 	//printf("CVFD::ShowIcon %s %x\n", show ? "show" : "hide", (int) icon);
 	struct vfd_ioctl_data data;
 	memset(&data, 0, sizeof(struct vfd_ioctl_data));
@@ -1141,25 +1624,47 @@
 #ifdef HAVE_DUCKBOX_HARDWARE
 void CVFD::ClearIcons()
 {
-#if defined (BOXMODEL_ATEVIO7500)
+#if defined (BOXMODEL_HS7110) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_HS7819) \
+ || defined (BOXMODEL_CUBEREVO_MINI_FTA) \
+ || defined (BOXMODEL_CUBEREVO_250HD) \
+ || defined (BOXMODEL_IPBOX55) \
+ || defined (BOXMODEL_ATEMIO520) \
+ || defined (BOXMODEL_OPT9600) \
+ || defined (BOXMODEL_OPT9600MINI)
 	return;
 #endif
-	for (int id = 0x10; id < FP_ICON_MAX; id++)
-	{
-#if defined (BOXMODEL_OCTAGON1008)
+	for (int id = 1; id < FP_ICON_MAX; id++)
+	{
+#if defined (BOXMODEL_HS9510) \
+ || defined (BOXMODEL_UFS912) \
+ || defined (BOXMODEL_UFS913) \
+ || defined (BOXMODEL_UFS922)
 		if (id != FP_ICON_USB && id != FP_ICON_HDD)
-#elif defined(BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_TF7700)
+#elif defined(BOXMODEL_FS9000) \
+ ||   defined (BOXMODEL_TF7700)
 		if (id != FP_ICON_USB)
 #else
 		if (id != 0x10 && id != 0x12)
 #endif
+		{
 			ShowIcon((fp_icon)id, false);
+		}
 	}
 	return;
 }
 
 void CVFD::ShowText(const char *str)
 {
+	if (! str)
+	{
+		printf("CVFD::ShowText: str is NULL!\n");
+		return;
+	}
 	memset(g_str, 0, sizeof(g_str));
 	memcpy(g_str, str, sizeof(g_str) - 1);
 
@@ -1174,6 +1679,7 @@
 	}
 	ShowNormalText(g_str, false);
 }
+
 void CVFD::repaintIcons()
 {
 #if BOXMODEL_UFS912 || BOXMODEL_UFS913
@@ -1187,7 +1693,7 @@
 	}
 #endif
 }
-#else
+#else  // non-duckbox
 void CVFD::ShowText(const char *str)
 {
 	if (fd < 0 || !support_text)
@@ -1223,11 +1729,30 @@
 
 void CVFD::ShowNumber(int number)
 {
+	char buffer[VFDLENGTH + 1];
+	int n;
+
 	if (fd < 0 || (!support_text && !support_numbers))
-		return;
-
+	{
+		return;
+	}
 	if (number < 0)
-		return;
+	{
+		return;
+	}
+	memset (buffer, 0, sizeof(buffer));
+#if defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7819) \
+ || defined (BOXMODEL_CUBEREVO_MINI_FTA) \
+ || defined (BOXMODEL_CUBEREVO_250HD) \
+ || defined (BOXMODEL_ATEMIO520) \
+ || defined (BOXMODEL_OPT9600MINI)
+	n = sprintf(buffer, "%04d", number);
+#else
+	n = sprintf(buffer, "%4d", number);
+#endif
+	CVFD::ShowText(buffer);
 }
 
 #ifdef VFD_UPDATE
@@ -1254,17 +1779,27 @@
 void CVFD::showInfoBox(const char *const title, const char *const text, int autoNewline, int timer)
 {
 #ifdef HAVE_LCD
-	if (!has_lcd) return;
+	if (!has_lcd)
+	{
+		return;
+	}
 	//printf("[lcdd] Info: \n");
 	if (text != NULL)
+	{
 		m_infoBoxText = text;
+	}
 	if (text != NULL)
+	{
 		m_infoBoxTitle = title;
+	}
 	if (timer != -1)
+	{
 		m_infoBoxTimer = timer;
+	}
 	if (autoNewline != -1)
+	{
 		m_infoBoxAutoNewline = autoNewline;
-
+	}
 	//printf("[lcdd] Info: %s,%s,%d,%d\n",m_infoBoxTitle.c_str(),m_infoBoxText.c_str(),m_infoBoxAutoNewline,m_infoBoxTimer);
 	if (mode == MODE_INFOBOX &&
 		!m_infoBoxText.empty())
@@ -1279,7 +1814,9 @@
 		{
 			int width = fonts.menu->getRenderWidth(m_infoBoxTitle);
 			if (width > 100)
+			{
 				width = 100;
+			}
 			int start_pos = (120 - width) / 2;
 			display.draw_fill_rect(start_pos, EPG_INFO_WINDOW_POS - 4,       start_pos + width + 5,        EPG_INFO_WINDOW_POS + 10,    CLCDDisplay::PIXEL_OFF);
 			fonts.menu->RenderString(start_pos + 4, EPG_INFO_WINDOW_POS + 5, width + 5, m_infoBoxTitle.c_str(), CLCDDisplay::PIXEL_ON);
@@ -1314,39 +1851,49 @@
 /*****************************************************************************************/
 //showFilelist
 /*****************************************************************************************/
-#define BAR_POS_X               114
-#define BAR_POS_Y                10
-#define BAR_POS_WIDTH     6
-#define BAR_POS_HEIGTH   40
+#define BAR_POS_X     114
+#define BAR_POS_Y      10
+#define BAR_POS_WIDTH   6
+#define BAR_POS_HEIGTH 40
 
 void CVFD::showFilelist(int flist_pos, CFileList *flist, const char *const mainDir)
 {
 #ifdef HAVE_LCD
-	if (!has_lcd) return;
+	if (!has_lcd)
+	{
+		return;
+	}
 	//printf("[lcdd] FileList\n");
 	if (flist != NULL)
+	{
 		m_fileList = flist;
+	}
 	if (flist_pos != -1)
+	{
 		m_fileListPos = flist_pos;
+	}
 	if (mainDir != NULL)
+	{
 		m_fileListHeader = mainDir;
-
-	if (mode == MODE_FILEBROWSER &&
-		m_fileList != NULL &&
-		!m_fileList->empty())
-	{
-
+	}
+	if (mode == MODE_FILEBROWSER
+	&&  m_fileList != NULL
+ 	&&  !m_fileList->empty())
+	{
 		printf("[lcdd] FileList:OK\n");
 		int size = m_fileList->size();
 
 		display.draw_fill_rect(-1, -1, 120, 52, CLCDDisplay::PIXEL_OFF); // clear lcd
 
 		if (m_fileListPos > size)
+		{
 			m_fileListPos = size - 1;
-
+		}
 		int width = fonts.menu->getRenderWidth(m_fileListHeader);
 		if (width > 110)
+		{
 			width = 110;
+		}
 		fonts.menu->RenderString((120 - width) / 2, 11, width + 5, m_fileListHeader.c_str(), CLCDDisplay::PIXEL_ON);
 
 		//printf("list%d,%d\r\n",m_fileListPos,(*m_fileList)[m_fileListPos].Marked);
@@ -1401,7 +1948,9 @@
 		int pages = (((size - 1) / 3) + 1);
 		int marker_length = (BAR_POS_HEIGTH - 2) / pages;
 		if (marker_length < 4)
+		{
 			marker_length = 4; // not smaller than 4 pixel
+		}
 		int marker_offset = ((BAR_POS_HEIGTH - 2 - marker_length) * m_fileListPos) / size  ;
 		//printf("%d,%d,%d\r\n",pages,marker_length,marker_offset);
 
@@ -1424,24 +1973,34 @@
 void CVFD::showProgressBar(int global, const char *const text, int show_escape, int timer)
 {
 #ifdef HAVE_LCD
-	if (!has_lcd) return;
+	if (!has_lcd)
+	{
+		return;
+	}
 	if (text != NULL)
+	{
 		m_progressHeaderGlobal = text;
-
+	}
 	if (timer != -1)
+	{
 		m_infoBoxTimer = timer;
-
+	}
 	if (global >= 0)
 	{
 		if (global > 100)
+		{
 			m_progressGlobal = 100;
+		}
 		else
+		{
 			m_progressGlobal = global;
+		}
 	}
 
 	if (show_escape != -1)
+	{
 		m_progressShowEscape = show_escape;
-
+	}
 	if (mode == MODE_PROGRESSBAR)
 	{
 		//printf("[lcdd] prog:%s,%d,%d\n",m_progressHeaderGlobal.c_str(),m_progressGlobal,m_progressShowEscape);
@@ -1451,7 +2010,9 @@
 		// paint progress header
 		int width = fonts.menu->getRenderWidth(m_progressHeaderGlobal);
 		if (width > 100)
+		{
 			width = 100;
+		}
 		int start_pos = (120 - width) / 2;
 		fonts.menu->RenderString(start_pos, 12 + 12, width + 10, m_progressHeaderGlobal.c_str(), CLCDDisplay::PIXEL_ON);
 
@@ -1487,34 +2048,48 @@
 void CVFD::showProgressBar2(int local, const char *const text_local, int global, const char *const text_global, int show_escape)
 {
 #ifdef HAVE_LCD
-	if (!has_lcd) return;
+	if (!has_lcd)
+	{
+		return;
+	}
 	//printf("[lcdd] prog2\n");
 	if (text_local != NULL)
+	{
 		m_progressHeaderLocal = text_local;
-
+	}
 	if (text_global != NULL)
+	{
 		m_progressHeaderGlobal = text_global;
-
+	}
 	if (global >= 0)
 	{
 		if (global > 100)
+		{
 			m_progressGlobal = 100;
+		}
 		else
+		{
 			m_progressGlobal = global;
+		}
 	}
 	if (local >= 0)
 	{
 		if (local > 100)
+		{
 			m_progressLocal = 100;
+		}
 		else
+		{
 			m_progressLocal = local;
+		}
 	}
 	if (show_escape != -1)
+	{
 		m_progressShowEscape = show_escape;
+	}
 
 	if (mode == MODE_PROGRESSBAR2)
 	{
-
 		//printf("[lcdd] prog2:%s,%d,%d\n",m_progressHeaderGlobal.c_str(),m_progressGlobal,m_progressShowEscape);
 		// Clear Display
 		display.draw_fill_rect(0, 12, 120, 64, CLCDDisplay::PIXEL_OFF);
@@ -1522,7 +2097,9 @@
 		// paint  global header
 		int width = fonts.menu->getRenderWidth(m_progressHeaderGlobal);
 		if (width > 100)
+		{
 			width = 100;
+		}
 		int start_pos = (120 - width) / 2;
 		fonts.menu->RenderString(start_pos, PROG2_GLOB_POS_Y - 3, width + 10, m_progressHeaderGlobal.c_str(), CLCDDisplay::PIXEL_ON);
 
@@ -1535,7 +2112,9 @@
 		// paint  local header
 		width = fonts.menu->getRenderWidth(m_progressHeaderLocal);
 		if (width > 100)
+		{
 			width = 100;
+		}
 		start_pos = (120 - width) / 2;
 		fonts.menu->RenderString(start_pos, PROG2_LOCAL_POS_Y + PROG2_LOCAL_POS_HEIGTH + 10, width + 10, m_progressHeaderLocal.c_str(), CLCDDisplay::PIXEL_ON);
 		// paint local bar
@@ -1554,5 +2133,4 @@
 }
 /*****************************************************************************************/
 #endif // VFD_UPDATE
-
-
+// vim:ts=4
--- neutrino-ddt.org/src/gui/miscsettings_menu.cpp
+++ neutrino-ddt/src/gui/miscsettings_menu.cpp
@@ -345,7 +345,10 @@
 	{
 		if (fanNotifier == NULL)
 			fanNotifier = new CFanControlNotifier();
-#if defined (BOXMODEL_IPBOX9900) || defined (BOXMODEL_IPBOX99) || defined (BOXMODEL_DM8000)
+#if defined (BOXMODEL_IPBOX9900) \
+ || defined (BOXMODEL_IPBOX99) \
+ || defined (BOXMODEL_CUBEREVO) \
+ || defined (BOXMODEL_CUBEREVO_9500HD)
 		CMenuOptionNumberChooser * mn = new CMenuOptionNumberChooser(LOCALE_FAN_SPEED, &g_settings.fan_speed, true, 0, 1, fanNotifier, CRCInput::RC_nokey, NULL, 0, 0, LOCALE_OPTIONS_OFF);
 #else
 		CMenuOptionNumberChooser * mn = new CMenuOptionNumberChooser(LOCALE_FAN_SPEED, &g_settings.fan_speed, true, 1, 14, fanNotifier, CRCInput::RC_nokey, NULL, 0, 0, LOCALE_OPTIONS_OFF);
--- neutrino-ddt.org/src/gui/movieplayer.cpp
+++ neutrino-ddt/src/gui/movieplayer.cpp
@@ -512,7 +512,7 @@
  && !defined(BOXMODEL_UFS912) \
  && !defined(BOXMODEL_UFS913) \
  && !defined(BOXMODEL_UFS922) \
- && !defined(BOXMODEL_OCTAGON1008) \
+ && !defined(BOXMODEL_HS9510) \
  && !defined(BOXMODEL_IPBOX9900) \
  && !defined(BOXMODEL_IPBOX99) \
  && !defined(BOXMODEL_IPBOX55)
@@ -521,7 +521,7 @@
 				lcd = "";
 #endif
 			break;
-#if !defined(BOXMODEL_OCTAGON1008)
+#if !defined(BOXMODEL_HS9510)
 		case CMoviePlayerGui::REW:
 			sprintf(tmp, "%dx<< ", abs(speed));
 			lcd = tmp;
@@ -536,8 +536,8 @@
  && !defined(BOXMODEL_UFS912) \
  && !defined(BOXMODEL_UFS913) \
  && !defined(BOXMODEL_UFS922) \
- && !defined(BOXMODEL_FORTIS_HDBOX) \
- && !defined(BOXMODEL_OCTAGON1008) \
+ && !defined(BOXMODEL_FS9000) \
+ && !defined(BOXMODEL_HS9510) \
  && !defined(BOXMODEL_CUBEREVO_MINI) \
  && !defined(BOXMODEL_CUBEREVO_MINI2) \
  && !defined(BOXMODEL_CUBEREVO_3000HD) \
--- neutrino-ddt.org/src/gui/update.cpp
+++ neutrino-ddt/src/gui/update.cpp
@@ -96,7 +96,7 @@
 #define UPDATE_LOCAL_FILENAME          "update.img"
 #define RELEASE_CYCLE                  "2.0"
 #define FILEBROWSER_UPDATE_FILTER      "img"
-#if BOXMODEL_UFS910 || BOXMODEL_FORTIS_HDBOX || BOXMODEL_OCTAGON1008
+#if BOXMODEL_UFS910 || BOXMODEL_FS9000 || BOXMODEL_HS9510
 #define MTD_OF_WHOLE_IMAGE              5
 #define MTD_DEVICE_OF_UPDATE_PART       "/dev/mtd5"
 #elif BOXMODEL_CUBEREVO || BOXMODEL_CUBEREVO_MINI || BOXMODEL_CUBEREVO_MINI2
--- neutrino-ddt.org/src/gui/widget/Makefile.am
+++ neutrino-ddt/src/gui/widget/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -fno-rtti -fno-exceptions -D__STDC_FORMAT_MACROS
+AM_CPPFLAGS = -fno-rtti -fno-exceptions -D__STDC_FORMAT_MACROS -std=c++11
 
 AM_CPPFLAGS += \
 	-I$(top_builddir) \
--- neutrino-ddt.org/src/system/setting_helpers.cpp
+++ neutrino-ddt/src/system/setting_helpers.cpp
@@ -625,10 +625,20 @@
 	int cfd;
 
 	printf("FAN Speed %d\n", speed);
-#if defined (BOXMODEL_IPBOX9900) || defined (BOXMODEL_IPBOX99)
+#if defined (BOXMODEL_IPBOX9900) \
+ || defined (BOXMODEL_IPBOX99)
 	cfd = open("/proc/stb/misc/fan", O_WRONLY);
 	if(cfd < 0) {
 		perror("Cannot open /proc/stb/misc/fan");
+#elif defined (BOXMODEL_CUBEREVO) \
+ ||   defined (BOXMODEL_CUBEREVO_9500HD)
+	cfd = open("/proc/stb/fp/fan", O_WRONLY);
+	if(cfd < 0) {
+		perror("Cannot open /proc/stb/fp/fan");
+#elif defined (BOXMODEL_UFS922)
+	cfd = open("/proc/stb/fp/fan_pwm", O_WRONLY);
+	if(cfd < 0) {
+		perror("Cannot open /proc/stb/fp/fan_pwm");
 #else
 	cfd = open("/proc/stb/fan/fan_ctrl", O_WRONLY);
 	if(cfd < 0) {
@@ -639,7 +649,10 @@
 
 	switch (speed)
 
-#if defined (BOXMODEL_IPBOX9900) || defined (BOXMODEL_IPBOX99)
+#if defined (BOXMODEL_IPBOX9900) \
+ || defined (BOXMODEL_IPBOX99) \
+ || defined (BOXMODEL_CUBEREVO) \
+ || defined (BOXMODEL_CUBEREVO_9500HD)
 	{
 	case 0:
 		write(cfd,"0",1);
@@ -648,6 +661,18 @@
 		write(cfd,"1",1);
 		break;
 	}
+#elif defined (BOXMODEL_UFS922)
+	{
+	case 0:
+		write(cfd,"0",1);
+		break;
+	case 1:
+		write(cfd,"128",1);
+		break;
+	case 2:
+		write(cfd,"255",1);
+		break;
+	}
 #else
 	{
 	case 1:

