# commit f5f2e6e066e99323695989f5cdf606b67256481d # 23/10/2019
--- neutrino-mp-tangos.org/acinclude.m4
+++ neutrino-mp-tangos/acinclude.m4
@@ -435,7 +435,7 @@
 AC_ARG_WITH(boxtype,
 	AS_HELP_STRING([--with-boxtype], [valid values: tripledragon, coolstream, spark, azbox, generic, armbox, duckbox, spark7162, mipsbox]),
 	[case "${withval}" in
-		tripledragon|coolstream|azbox|generic|armbox)
+		tripledragon|coolstream|azbox|generic|armbox|duckbox)
 			BOXTYPE="$withval"
 		;;
 		spark|spark7162)
@@ -446,50 +446,6 @@
 			BOXTYPE="dreambox"
 			BOXMODEL="$withval"
 		;;
-		ufs*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		atevio*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		fortis*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		octagon*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		hs7*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		dp*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		cuberevo*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		ipbox*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		arivalink200)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		tf*)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
-		hl101)
-			BOXTYPE="duckbox"
-			BOXMODEL="$withval"
-		;;
 		hd51|hd60|hd61|bre2ze4k|vusolo4k|vuduo4k|vuultimo4k|vuzero4k|vuuno4kse|vuuno4k)
 			BOXTYPE="armbox"
 			BOXMODEL="$withval"
@@ -506,8 +462,8 @@
 
 AC_ARG_WITH(boxmodel,
 	AS_HELP_STRING([--with-boxmodel], [valid for coolstream: hd1, hd2])
-AS_HELP_STRING([], [valid for duckbox: ufs910, ufs912, ufs913, ufs922, atevio7500, fortis_hdbox, octagon1008, hs7110, hs7810a, hs7119, hs7819, dp7000, cuberevo, cuberevo_mini, cuberevo_mini2, cuberevo_250hd, cuberevo_2000hd, cuberevo_3000hd, ipbox9900, ipbox99, ipbox55, arivalink200, tf7700, hl101])
 AS_HELP_STRING([], [valid for spark: spark, spark7162])
+AS_HELP_STRING([], [valid for duckbox: ufs910, ufs912, ufs913, ufs922, atevio7500, fortis_hdbox, octagon1008, hs7110, hs7420, hs7810a, hs7119, hs7429, hs7819, cuberevo, cuberevo_mini, cuberevo_mini2, cuberevo_250hd, cuberevo_2000hd, cuberevo_3000hd, ipbox9900, ipbox99, ipbox55, arivalink200, tf7700, hl101, adb_box, vip1_v1, vip1_v2, vip2, vitamin_hd5000])
 AS_HELP_STRING([], [valid for armbox: hd51, hd60, hd61, bre2ze4k, zgemmah7, vusolo4k, vuduo4k, vuultimo4k, vuzero4k, vuuno4kse, vuuno4k])
 AS_HELP_STRING([], [valid for mipsbox: vuduo])
 AS_HELP_STRING([], [valid for generic: raspi]),
@@ -531,7 +487,7 @@
 				AC_MSG_ERROR([unknown model $withval for boxtype $BOXTYPE])
 			fi
 		;;
-		ufs910|ufs912|ufs913|ufs922|atevio7500|fortis_hdbox|octagon1008|hs7110|hs7810a|hs7119|hs7819|dp7000|cuberevo|cuberevo_mini|cuberevo_mini2|cuberevo_250hd|cuberevo_2000hd|cuberevo_3000hd|ipbox9900|ipbox99|ipbox55|arivalink200|tf7700|hl101)
+		ufs910|ufs912|ufs913|ufs922|atevio7500|fortis_hdbox|octagon1008|hs7110|hs7420|hs7810a|hs7119|hs7429|hs7819|dp7000|cuberevo|cuberevo_mini|cuberevo_mini2|cuberevo_250hd|cuberevo_2000hd|cuberevo_3000hd|ipbox9900|ipbox99|ipbox55|arivalink200|tf7700|hl101|adb_box|vip1_v1|vip1_v2|vip2|vitamin_hd5000)
 			if test "$BOXTYPE" = "duckbox"; then
 				BOXMODEL="$withval"
 			else
@@ -596,8 +552,10 @@
 AM_CONDITIONAL(BOXMODEL_FORTIS_HDBOX, test "$BOXMODEL" = "fortis_hdbox")
 AM_CONDITIONAL(BOXMODEL_OCTAGON1008, test "$BOXMODEL" = "octagon1008")
 AM_CONDITIONAL(BOXMODEL_HS7110, test "$BOXMODEL" = "hs7110")
+AM_CONDITIONAL(BOXMODEL_HS7420,test "$BOXMODEL" = "hs7420")
 AM_CONDITIONAL(BOXMODEL_HS7810A, test "$BOXMODEL" = "hs7810a")
 AM_CONDITIONAL(BOXMODEL_HS7119, test "$BOXMODEL" = "hs7119")
+AM_CONDITIONAL(BOXMODEL_HS7429,test "$BOXMODEL" = "hs7429")
 AM_CONDITIONAL(BOXMODEL_HS7819, test "$BOXMODEL" = "hs7819")
 AM_CONDITIONAL(BOXMODEL_DP7000, test "$BOXMODEL" = "dp7000")
 
@@ -613,6 +571,11 @@
 AM_CONDITIONAL(BOXMODEL_ARIVALINK200, test "$BOXMODEL" = "arivalink200")
 AM_CONDITIONAL(BOXMODEL_TF7700, test "$BOXMODEL" = "tf7700")
 AM_CONDITIONAL(BOXMODEL_HL101, test "$BOXMODEL" = "hl101")
+AM_CONDITIONAL(BOXMODEL_ADB_BOX, test "$BOXMODEL" = "adb_box")
+AM_CONDITIONAL(BOXMODEL_VIP1_V1, test "$BOXMODEL" = "vip1_v1")
+AM_CONDITIONAL(BOXMODEL_VIP1_V2, test "$BOXMODEL" = "vip1_v2")
+AM_CONDITIONAL(BOXMODEL_VIP2, test "$BOXMODEL" = "vip2")
+AM_CONDITIONAL(BOXMODEL_VITAMIN_HD5000, test "$BOXMODEL" = "vitamin_hd5000")
 
 AM_CONDITIONAL(BOXMODEL_HD51, test "$BOXMODEL" = "hd51")
 AM_CONDITIONAL(BOXMODEL_HD60, test "$BOXMODEL" = "hd60")
@@ -676,10 +639,14 @@
 	AC_DEFINE(BOXMODEL_OCTAGON1008, 1, [octagon1008])
 elif test "$BOXMODEL" = "hs7110"; then
 	AC_DEFINE(BOXMODEL_HS7110, 1, [hs7110])
+elif test "$BOXMODEL" = "hs7420"; then
+	AC_DEFINE(BOXMODEL_HS7420, 1, [hs7420])
 elif test "$BOXMODEL" = "hs7810a"; then
 	AC_DEFINE(BOXMODEL_HS7810A, 1, [hs7810a])
 elif test "$BOXMODEL" = "hs7119"; then
 	AC_DEFINE(BOXMODEL_HS7119, 1, [hs7119])
+elif test "$BOXMODEL" = "hs7429"; then
+	AC_DEFINE(BOXMODEL_HS7429, 1, [hs7429])
 elif test "$BOXMODEL" = "hs7819"; then
 	AC_DEFINE(BOXMODEL_HS7819, 1, [hs7819])
 elif test "$BOXMODEL" = "dp7000"; then
@@ -708,6 +674,16 @@
 	AC_DEFINE(BOXMODEL_TF7700, 1, [tf7700])
 elif test "$BOXMODEL" = "hl101"; then
 	AC_DEFINE(BOXMODEL_HL101, 1, [hl101])
+elif test "$BOXMODEL" = "adb_box"; then
+	AC_DEFINE(BOXMODEL_ADB_BOX, 1, [adb_box])
+elif test "$BOXMODEL" = "vip1_v1"; then
+	AC_DEFINE(BOXMODEL_VIP1_V1, 1, [vip1_v1])
+elif test "$BOXMODEL" = "vip1_v2"; then
+	AC_DEFINE(BOXMODEL_VIP1_V2, 1, [vip1_v2])
+elif test "$BOXMODEL" = "vip2"; then
+	AC_DEFINE(BOXMODEL_VIP2, 1, [vip2])
+elif test "$BOXMODEL" = "vitamin_hd5000"; then
+	AC_DEFINE(BOXMODEL_VITAMIN_HD5000, 1, [vitamin_hd5000])
 elif test "$BOXMODEL" = "hd51"; then
 	AC_DEFINE(BOXMODEL_HD51, 1, [hd51])
 	AC_DEFINE(ENABLE_CHANGE_OSD_RESOLUTION, 1, [enable change the osd resolution])
--- neutrino-mp-tangos.org/src/neutrino.cpp
+++ neutrino-mp-tangos/src/neutrino.cpp
@@ -4760,6 +4760,9 @@
 
 		CVFD::getInstance()->Clear();
 		CVFD::getInstance()->setMode(CVFD::MODE_STANDBY);
+#if defined (FORTIS_HDBOX) || defined (TF7700)
+		CVFD::getInstance()->setlcdparameter(g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS], 0);
+#endif
 		CVFD::getInstance()->setBacklight(g_settings.backlight_standby);
 
 		InfoClock->enableInfoClock(false);
@@ -4795,7 +4798,13 @@
 		// Active standby off
 		powerManager->SetStandby(false, false);
 		CVFD::getInstance()->setMode(CVFD::MODE_TVRADIO);
-		CVFD::getInstance()->ShowText("resume");
+#if defined (FORTIS_HDBOX) || defined (TF7700)
+		CVFD::getInstance()->setlcdparameter(g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS], 1);
+#endif
+#ifdef HAVE_DUCKBOX_HARDWARE
+		CVFD::getInstance()->ClearIcons();
+#endif
+		CVFD::getInstance()->ShowText("Resume");
 		cpuFreq->SetCpuFreq(g_settings.cpufreq * 1000 * 1000);
 		videoDecoder->Standby(false);
 		CEpgScan::getInstance()->Stop();
--- neutrino-mp-tangos.org/src/driver/vfd.cpp
+++ neutrino-mp-tangos/src/driver/vfd.cpp
@@ -52,27 +52,43 @@
 #include <stropts.h>
 #define VFD_DEVICE "/dev/vfd"
 
-#if defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_TF7700)
+#if defined (BOXMODEL_OCTAGON1008) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_TF7700) \
+ || defined (BOXMODEL_HL101) \
+ || defined (BOXMODEL_VIP1_V1) \
+ || defined (BOXMODEL_VIP1_V2) \
+ || defined (BOXMODEL_VIP2)
 #define VFDLENGTH 8
-#elif defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#elif defined (BOXMODEL_FORTIS_HDBOX) \
+ || defined (BOXMODEL_ATEVIO7500)
 #define VFDLENGTH 12
-#elif defined (BOXMODEL_HS7810A) || defined (BOXMODEL_HS7119) || defined (BOXMODEL_HS7819) || defined (BOXMODEL_CUBEREVO_250HD) || defined (BOXMODEL_IPBOX55)
+#elif defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7819) \
+ || defined (BOXMODEL_CUBEREVO_250HD) \
+ || defined (BOXMODEL_IPBOX55)
 #define VFDLENGTH 4
 #elif defined (BOXMODEL_HS7110)
 #define VFDLENGTH 0
-#elif defined (BOXMODEL_IPBOX9900) || defined (BOXMODEL_IPBOX99)
+#elif defined (BOXMODEL_IPBOX9900) \
+ || defined (BOXMODEL_IPBOX99)
 #define VFDLENGTH 14
+#elif defined (BOXMODEL_ADB_BOX)
+#define VFDLENGTH 15
 #else
 #define VFDLENGTH 16
 #endif
 
-#define SCROLL_TIME 100000
+#define SCROLL_TIME 500000
 
 bool invert = false;
 bool blocked = false;
 int blocked_counter = 0;
 int file_vfd = -1;
 bool active_icon[16] = { false };
+int adb_led = -1;
 
 pthread_t vfd_scrollText;
 
@@ -81,6 +97,33 @@
 	unsigned char data[64];
 	unsigned char length;
 };
+
+#if defined (BOXMODEL_ADB_BOX)
+int get_adb_variant()
+{
+	FILE *adb_file;
+	char adb_model[5] = { 0x00 };
+	int  ret;
+
+	adb_file = fopen("/proc/stb/info/adb_variant", "r");
+	if (adb_file != NULL)
+	{
+		fread(adb_model, 4, 1, adb_file);
+		printf("[CVFD] get_adb_variant: detected ADB model %s\n", adb_model);
+		if (strcmp(adb_model, "bska") == 0 || strcmp(adb_model, "bxzb") == 0)
+		{
+			printf("[CVFD] get_adb_variant: LED display assumed.\n");
+			ret = 1;
+		}
+		else
+		{
+			printf("[CVFD] get_adb_variant: VFD display assumed.\n");
+			ret = 0;
+		}
+	}
+	return ret;
+}
+#endif
 
 static void write_to_vfd(unsigned int DevType, struct vfd_ioctl_data * data, bool force = false)
 {
@@ -119,7 +162,8 @@
 	}
 }
 
-#if defined (BOXMODEL_UFS910) || defined (BOXMODEL_UFS922)
+#if defined (BOXMODEL_UFS910) \
+ || defined (BOXMODEL_UFS922)
 static void writeCG (unsigned char adress, unsigned char pixeldata[5])
 {
 	struct vfd_ioctl_data data;
@@ -137,7 +181,16 @@
 
 static void ShowNormalText(char * str, bool fromScrollThread = false)
 {
-	if (blocked)
+	int colon_at_2 = 0;
+	int disp_len = 0;
+
+#if defined (BOXMODEL_HS7119) || defined (BOXMODEL_HS7810A) || defined (BOXMODEL_HS7819)
+	if (str[2] == 0x3a)
+	{
+		colon_at_2++;
+	}
+#endif
+ 	if (blocked)
 	{
 		printf("[CVFD] - blocked\n");
 		usleep(SCROLL_TIME);
@@ -156,39 +209,49 @@
 			vfd_scrollText = 0;
 		}
 	}
-	if ((strlen(str) > VFDLENGTH && !fromScrollThread) && (g_settings.lcd_vfd_scroll >= 1))
+	disp_len = (adb_led == 1 ? 4 : VFDLENGTH);
+#if defined (BOXMODEL_HS7119) || defined (BOXMODEL_HS7810A) || defined (BOXMODEL_HS7819)
+	if ((strlen(str) > disp_len + colon_at_2 && !fromScrollThread) && (g_settings.lcd_vfd_scroll >= 1))
+#else
+	if ((strlen(str) > disp_len && !fromScrollThread) && (g_settings.lcd_vfd_scroll >= 1))
+#endif
 	{
 		CVFD::getInstance()->ShowScrollText(str);
 		return;
 	}
-
-	if (strlen(str) < VFDLENGTH && VFDLENGTH > 7) // do not center on small displays
-		ws = (VFDLENGTH-strlen(str))/2;
+	if (strlen(str) < disp_len && disp_len > 7) // do not center on small displays
+	{
+		ws = (disp_len - strlen(str)) / 2;
+	}
 	else
+	{
 		ws = 0;
+	}
 	memset(data.data, ' ', 63);
+	data.start = 0;
 	if (!fromScrollThread)
 	{
-		memcpy (data.data+ws, str, VFDLENGTH-ws);
-		data.start = 0;
-		if ((strlen(str) % 2) == 1 && VFDLENGTH > 7) // do not center on small displays
-			data.length = VFDLENGTH-ws-1;
-		else
-			data.length = VFDLENGTH-ws;
+		memcpy(data.data + ws, str, disp_len - ws + colon_at_2);
+//		if ((strlen(str) % 2) == 1 && disp_len > 7) // if text length odd
+//		{
+//			data.length = disp_len - ws - 1 + colon_at_2;
+//		}
+//		else
+//		{
+			data.length = disp_len - ws + colon_at_2;
+//		}
 	}
 	else
 	{
-		memcpy ( data.data, str, VFDLENGTH);
-		data.start = 0;
-		data.length = VFDLENGTH;
-	}
+		memcpy(data.data, str, disp_len + colon_at_2);
+		data.length = disp_len + colon_at_2;
+	}
+//	printf("CVFD::ShowNormalText: [%s]\n", str);
 	write_to_vfd(VFDDISPLAYCHARS, &data);
 	return;
 }
 void CVFD::ShowScrollText(char *str)
 {
-	printf("CVFD::ShowScrollText: [%s]\n", str);
-
 	if (blocked)
 	{
 		printf("[CVFD] - blocked\n");
@@ -217,17 +280,18 @@
 	int i;
 	char *str = (char *)arg;
 	int len = strlen(str);
-	char out[VFDLENGTH+1];
-	char buf[VFDLENGTH+65];
-
-	memset(out, 0, VFDLENGTH+1);
+	char out[17];
+	char buf[81];
+	int disp_len = adb_led == 1 ? 4 : VFDLENGTH;
+
+	memset(out, 0, disp_len + 1);
 
 	int retries = g_settings.lcd_vfd_scroll;
 
-	if (len > VFDLENGTH)
+	if (len > disp_len)
 	{
 		printf("CVFD::ThreadScrollText: [%s], length %d\n", str, len);
-		memset(buf, ' ', (len + VFDLENGTH));
+		memset(buf, ' ', len + disp_len);
 		memcpy(buf, str, len);
 
 		while(retries--)
@@ -237,13 +301,13 @@
 			for (i = 0; i <= (len-1); i++)
 			{
 				// scroll text until end
-				memcpy(out, buf+i, VFDLENGTH);
+				memcpy(out, buf + i, disp_len);
 				ShowNormalText(out,true);
 				usleep(SCROLL_TIME);
 			}
 		}
 	}
-	memcpy(out, str, VFDLENGTH); // display first VFDLENGTH chars after scrolling
+	memcpy(out, str, disp_len); // display first VFDLENGTH chars after scrolling
 	ShowNormalText(out,true);
 
 	pthread_exit(0);
@@ -341,47 +405,62 @@
 	return lcdd;
 }
 
-void CVFD::count_down() {
-	if (timeout_cnt > 0) {
+void CVFD::count_down()
+{
+	if (timeout_cnt > 0)
+	{
 		timeout_cnt--;
-		if (timeout_cnt == 0 ) {
-			if (g_settings.lcd_setting_dim_brightness > -1) {
+		if (timeout_cnt == 0 )
+		{
+			if (g_settings.lcd_setting_dim_brightness > -1)
+			{
 				// save lcd brightness, setBrightness() changes global setting
 				int b = g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS];
 				setBrightness(g_settings.lcd_setting_dim_brightness);
 				g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS] = b;
-			} else {
+			}
+			else
+			{
 				setPower(0);
 			}
 		}
 	}
-	if (g_settings.lcd_info_line && switch_name_time_cnt > 0) {
+	if (g_settings.lcd_info_line && switch_name_time_cnt > 0)
+	{
 	  switch_name_time_cnt--;
-		if (switch_name_time_cnt == 0) {
-			if (g_settings.lcd_setting_dim_brightness > -1) {
+		if (switch_name_time_cnt == 0)
+		{
+			if (g_settings.lcd_setting_dim_brightness > -1)
+			{
 				CVFD::getInstance()->showTime(true);
 			}
 		}
 	}
 }
 
-void CVFD::wake_up() {
-	if(fd < 0) return;
-
-	if (atoi(g_settings.lcd_setting_dim_time.c_str()) > 0) {
+void CVFD::wake_up()
+{
+	if (fd < 0)
+	{
+		return;
+	}
+	if (atoi(g_settings.lcd_setting_dim_time.c_str()) > 0)
+	{
 		timeout_cnt = atoi(g_settings.lcd_setting_dim_time.c_str());
 		g_settings.lcd_setting_dim_brightness > -1 ?
 			setBrightness(g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS]) : setPower(1);
 	}
 	else
+	{
 		setPower(1);
-	if(g_settings.lcd_info_line){
+	}
+	if (g_settings.lcd_info_line)
+	{
 		switch_name_time_cnt = g_settings.handling_infobar[SNeutrinoSettings::HANDLING_INFOBAR] + 10;
 	}
 #if defined (BOXMODEL_OCTAGON1008)
 	ShowIcon(ICON_COLON2, false);
 #endif
-
 }
 
 void* CVFD::TimeThread(void *)
@@ -389,17 +468,24 @@
 	while(1) {
 		sleep(1);
 		struct stat buf;
-                if (stat("/tmp/vfd.locked", &buf) == -1) {
-                        CVFD::getInstance()->showTime();
-                        CVFD::getInstance()->count_down();
-                } else
-                        CVFD::getInstance()->wake_up();
+                if (stat("/tmp/vfd.locked", &buf) == -1)
+		{
+                	CVFD::getInstance()->showTime();
+                	CVFD::getInstance()->count_down();
+                }
+		else
+		{
+                	CVFD::getInstance()->wake_up();
+		}
 	}
 	return NULL;
 }
 
 void CVFD::init(const char * /*fontfile*/, const char * /*fontname*/)
 {
+#if defined (BOXMODEL_ADB_BOX)
+	adb_led = get_adb_variant();
+#endif		
 	//InitNewClock(); /FIXME
 
 	brightness = -1;
@@ -428,7 +514,7 @@
 
 	brightness = dimm;
 
-printf("CVFD::setlcdparameter dimm %d power %d\n", dimm, power);
+	printf("CVFD::setlcdparameter dimm %d power %d\n", dimm, power);
 #if !HAVE_DUCKBOX_HARDWARE
 	int ret = ioctl(fd, IOC_FP_SET_BRIGHT, dimm);
 	if(ret < 0)
@@ -436,51 +522,81 @@
 #else
 // Brightness
 	struct vfd_ioctl_data data;
-#if !defined (BOXMODEL_HS7810A) && !defined (BOXMODEL_HS7119) && !defined (BOXMODEL_HS7819)
 	memset(&data, 0, sizeof(struct vfd_ioctl_data));
 	data.start = brightness & 0x07;
 	data.length = 0;
 	write_to_vfd(VFDBRIGHTNESS, &data);
-#endif
-#if defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#if defined (BOXMODEL_FORTIS_HDBOX) \
+ || defined (BOXMODEL_ATEVIO7500)
 	usleep(100000);
 	memset(&data, 0, sizeof(struct vfd_ioctl_data));
-	data.start = 0;
+	if (power) {
+		data.start = 0x01; // red led
+	}
+	else
+	{
+		data.start = 0xf2; // cross plus blue led
+	}
+	data.data[3] = 0; // off
+	data.data[0] = 0;
+	data.data[1] = 0;
+	data.data[4] = 0;
 	data.length = 5;
-	if (power) {
-		data.data[0] = 0x01; // red led
-	}
-	else
-	{
-		data.data[0] = 0xf2; // cross plus blue led
-	}
-	data.start = 0;
-	data.data[4] = 0; // off
-	data.length = 5;
-	write_to_vfd(VFDPWRLED, &data);
+	write_to_vfd(VFDSETLED, &data);
 	usleep(100000);
 	memset(&data, 0, sizeof(struct vfd_ioctl_data));
-	data.start = 0;
+	if (power) {
+		data.start = 0xf2; // cross plus blue led
+	}
+	else
+	{
+		data.start = 0x01; // red led
+	}
+	data.data[3] = brightness * 2;
+	data.data[0] = 0;
+	data.data[1] = 0;
+	data.data[4] = 0;
 	data.length = 5;
-	if (power) {
-		data.data[0] = 0xf2; // cross plus blue led
+	write_to_vfd(VFDSETLED, &data);
+#elif defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_HS7819)
+	memset(&data, 0, sizeof(struct vfd_ioctl_data));
+	data.start = 0x02; // logo
+	data.data[3] = (brightness & 0x07);
+	data.data[0] = 0;
+	data.data[1] = 0;
+	data.data[4] = 0;
+	data.length = 5;
+	write_to_vfd(VFDSETLED, &data);
+#elif defined (BOXMODEL_ADB_BOX)
+	if (adb_led)
+	{
+		memset(&data, 0, sizeof(struct vfd_ioctl_data));
+		data.start = brightness & 0x07;
+		data.length = 0;
+		write_to_vfd(VFDLEDBRIGHTNESS, &data);
+	}
+	memset(&data, 0, sizeof(struct vfd_ioctl_data));
+	data.start = 0x01; // power LED
+	data.data[0] = 0;
+	data.data[1] = 0;
+	data.data[4] = 0;
+	data.length = 5;
+	if (power)
+	{
+		data.data[3] = 2;  // green
 	}
 	else
 	{
-		data.data[0] = 0x01; // red led
-	}
-	data.start = 0;
-	data.data[4] = brightness*2;
-	data.length = 5;
-	write_to_vfd(VFDPWRLED, &data);
-#elif defined (BOXMODEL_HS7810A) || defined (BOXMODEL_HS7819)
-	memset(&data, 0, sizeof(struct vfd_ioctl_data));
-	data.start = 0;
-	data.data[0] = 0x02; // logo
-	data.data[4] = (brightness & 0x07);
-	data.length = 5;
-	write_to_vfd(VFDPWRLED, &data);
-#elif !defined (BOXMODEL_UFS912) && !defined (BOXMODEL_UFS913) && !defined (BOXMODEL_OCTAGON1008)
+		data.data[3] = 1;  // red
+	}
+	write_to_vfd(VFDSETLED, &data);
+
+#elif !defined (BOXMODEL_UFS912) \
+ && !defined (BOXMODEL_UFS913) \
+ && !defined (BOXMODEL_OCTAGON1008)
 // Power on/off
 	if (power) {
 		data.start = 0x01;
@@ -534,6 +650,10 @@
 		return;
 
 	int led1 = -1, led2 = -1;
+#if defined(BOXMODEL_ADB_BOX)
+	int led3 = -1, led4 = -1;
+#endif
+
 	if(on_off){//on
 		switch(g_settings.led_rec_mode) {
 			case 1:
@@ -565,16 +685,18 @@
 				break;
 	      }
 	}
-
 	setled(led1, led2);
 }
 
 void CVFD::setled(void)
 {
+	int led1 = -1, led2 = -1;
+#if defined(BOXMODEL_ADB_BOX)
+	int led3 = -1, led4 = -1;
+#endif
+	int select = 0;
+
 	if(fd < 0) return;
-
-	int led1 = -1, led2 = -1;
-	int select = 0;
 
 	if(mode == MODE_MENU_UTF8 || mode == MODE_TVRADIO  )
 		  select = g_settings.led_tv_mode;
@@ -601,99 +723,140 @@
 }
 #endif
 
-void CVFD::showServicename(const std::string & name, int number) // UTF-8
+void CVFD::showServicename(const std::string &name, int number) // UTF-8
 {
 	if(fd < 0) return;
 
-printf("CVFD::showServicename: %s\n", name.c_str());
+//	printf("CVFD::showServicename: %s\n", name.c_str());
 	servicename = name;
 	service_number = number;
 
 	if (mode != MODE_TVRADIO)
 		return;
 
-	if (support_text)
+	if (strlen(name.c_str()) < 1)
+	{
+		return;
+	}
+	if (((support_text) && (VFDLENGTH > 7)) || service_number < 0)
+	{
+		printf("CVFD::showServicename (text): [%s]\n", name.c_str());
 		ShowText(name.c_str());
+	}
 	else
+	{
+		printf("CVFD::showServicename(number): [%d]\n", service_number);
 		ShowNumber(service_number);
-	wake_up();
+	}
 }
 
 void CVFD::showTime(bool force)
 {
 	//unsigned int system_rev = cs_get_revision();
 	static int recstatus = 0;
+
 #if 0
 	if(!has_lcd)
 		return;
 #endif
-	if(fd >= 0 && mode == MODE_SHUTDOWN) {
+	if (fd >= 0 && mode == MODE_SHUTDOWN)
+	{
 		ShowIcon(FP_ICON_CAM1, false);
 		return;
 	}
-	if (fd >= 0 && showclock) {
-		if (mode == MODE_STANDBY || ( g_settings.lcd_info_line && (MODE_TVRADIO == mode))) {
+	if (fd >= 0 && showclock)
+	{
+		if (mode == MODE_STANDBY || ( g_settings.lcd_info_line && (MODE_TVRADIO == mode)))
+		{
 			char timestr[21];
 			struct timeb tm;
 			struct tm * t;
-			static int hour = 0, minute = 0;
+			static int hour = 0, minute = 0, second = 0;
 
 			ftime(&tm);
 			t = localtime(&tm.time);
-			if(force || ( switch_name_time_cnt == 0 && ((hour != t->tm_hour) || (minute != t->tm_min))) ) {
+			if (force || ( switch_name_time_cnt == 0 && ((hour != t->tm_hour) || (minute != t->tm_min) || (second != t->tm_sec))) )
+			{
 				hour = t->tm_hour;
 				minute = t->tm_min;
-#if !defined (BOXMODEL_HS7810A) && !defined (BOXMODEL_HS7819)
+				second = t->tm_sec;
 #if defined (BOXMODEL_OCTAGON1008)
-				ShowIcon(ICON_COLON2, true);
-#elif defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_HS7119) || defined (BOXMODEL_CUBEREVO_250HD)
+
+				ShowIcon(ICON_COLON2, second % 2 == 0 ? true : false);
 				strftime(timestr, 5, "%H%M", t);
+				ShowText(timestr);
+#elif defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429)
+
+				ShowIcon(ICON_COLON2, second % 2 == 0 ? true : false);
+				strftime(timestr, 5, "%H%M", t);
+				ShowText(timestr);
+#elif defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7819)
+
+				strftime(timestr, 5, "%H%M", t);
+				ShowText(timestr);
+				ShowIcon(ICON_COLON, second % 2 == 0 ? true : false);
+#elif defined (BOXMODEL_FORTIS_HDBOX)
+				ShowIcon(ICON_COLON3, second % 2 == 0 ? true : false);
+				strftime(timestr, 13, "        %H%M", t);
+				ShowText(timestr);
+#elif defined (BOXMODEL_ADB_BOX)
+				if (adb_led)
+				{
+					strftime(timestr, 5, "%H%M", t);
+				}
+				else
+				{
+					strftime(timestr, 6, (second % 2 == 0 ? "%H:%M" : "%H %M"), t);
+				}
+				ShowText(timestr);
+#elif defined (BOXMODEL_CUBEREVO_250HD)
+				strftime(timestr, 5, "%H%M", t);
+				ShowText(timestr);
 #else
-				strftime(timestr, 6, "%H:%M", t);
-#endif
-				ShowText(timestr);
-#else //HS7810A or HS7819, string should not scroll
-				strftime(timestr, 6, "%H:%M", t);
-				struct vfd_ioctl_data data;
-				memset(data.data, ' ', 6);
-				memcpy (data.data, timestr, 6);
-				data.start = 0;
-				data.length = 5;
-				write_to_vfd(VFDDISPLAYCHARS, &data);
-#endif
-				if (support_text) {
+				if (support_text)
+				{
 					strftime(timestr, 20, "%H:%M", t);
 					ShowText(timestr);
-				} else if (support_numbers && has_led_segment) {
+				} else if (support_numbers && has_led_segment)
+				{
 					ShowNumber((t->tm_hour*100) + t->tm_min);
 #ifdef BOXMODEL_CS_HD2
 					ioctl(fd, IOC_FP_SET_COLON, 0x01);
 #endif
 				}
+#endif
 			}
 		}
 	}
-
 	int tmp_recstatus = CNeutrinoApp::getInstance()->recordingstatus;
-	if (tmp_recstatus) {
-		if(clearClock) {
+	if (tmp_recstatus)
+	{
+		if (clearClock)
+		{
 			clearClock = 0;
-			if(has_lcd)
+			if (has_lcd)
 				ShowIcon(FP_ICON_CAM1, false);
 #if !HAVE_DUCKBOX_HARDWARE
 			setled(false);//off
 #endif
-		} else {
+		}
+		else
+		{
 			clearClock = 1;
-			if(has_lcd)
+			if (has_lcd)
 				ShowIcon(FP_ICON_CAM1, true);
 #if !HAVE_DUCKBOX_HARDWARE
 			setled(true);//on
 #endif
 		}
-	} else if(clearClock || (recstatus != tmp_recstatus)) { // in case icon ON after record stopped
+	}
+	else if (clearClock || (recstatus != tmp_recstatus))
+	{ // in case icon ON after record stopped
 		clearClock = 0;
-		if(has_lcd)
+		if (has_lcd)
 			ShowIcon(FP_ICON_CAM1, false);
 
 #if !HAVE_DUCKBOX_HARDWARE
@@ -713,7 +876,9 @@
 		ShowIcon(FP_ICON_HD,chan->isHD());
 		ShowIcon(FP_ICON_LOCK,!chan->camap.empty());
 		if (chan->getAudioChannel() != NULL)
+		{
 			ShowIcon(FP_ICON_DD, chan->getAudioChannel()->audioChannelType == CZapitAudioChannel::AC3);
+		}
 	}
 }
 #endif
@@ -753,7 +918,8 @@
 		int pp = (int) round((double) vol / (double) 2);
 		if(oldpp != pp)
 		{
-#if defined (BOXMODEL_UFS910) || defined (BOXMODEL_UFS922)
+#if defined (BOXMODEL_UFS910) \
+ || defined (BOXMODEL_UFS922)
 			int i;
 			unsigned char speaker[5] = {0x1C, 0x1C, 0x1C, 0x3E, 0x7F}; // speaker symbol
 			writeCG(0, speaker);
@@ -800,15 +966,36 @@
 			char vol_chr[64] = "";
 			snprintf(vol_chr, sizeof(vol_chr)-1, "VOL: %d%%", (int)vol);
 			ShowText(vol_chr);
-#elif defined (BOXMODEL_OCTAGON1008)
+#elif defined (BOXMODEL_OCTAGON1008) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_HL101) \
+ || defined (BOXMODEL_VIP1_V1) \
+ || defined (BOXMODEL_VIP1_V2) \
+ || defined (BOXMODEL_VIP2)
 			char vol_chr[64] = "";
-			snprintf(vol_chr, sizeof(vol_chr)-1, "VOL=%3d", (int)vol);
+			snprintf(vol_chr, sizeof(vol_chr)-1, "VOL=%3d%%", (int)vol);
 			ShowText(vol_chr);
-#elif defined (BOXMODEL_HS7119) || defined (BOXMODEL_HS7810A) || defined (BOXMODEL_HS7819) || defined (BOXMODEL_CUBEREVO_250HD) || defined (BOXMODEL_IPBOX55)
+#elif defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7819) \
+ || defined (BOXMODEL_CUBEREVO_250HD) \
+ || defined (BOXMODEL_IPBOX55) \
+ || defined (BOXMODEL_ADB_BOX)
 			char vol_chr[64] = "";
 			snprintf(vol_chr, sizeof(vol_chr)-1, "v%3d", (int)vol);
 			ShowText(vol_chr);
-#elif defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500) || defined (BOXMODEL_UFS912) || defined (BOXMODEL_UFS913) || defined (BOXMODEL_CUBEREVO) || defined (BOXMODEL_CUBEREVO_MINI) || defined (BOXMODEL_CUBEREVO_MINI2) || defined (BOXMODEL_CUBEREVO_2000HD) || defined (BOXMODEL_CUBEREVO_3000HD) || defined (BOXMODEL_IPBOX9900) || defined (BOXMODEL_IPBOX99)
+#elif defined (BOXMODEL_FORTIS_HDBOX) \
+ || defined (BOXMODEL_ATEVIO7500) \
+ || defined (BOXMODEL_UFS912) \
+ || defined (BOXMODEL_UFS913) \
+ || defined (BOXMODEL_CUBEREVO) \
+ || defined (BOXMODEL_CUBEREVO_MINI) \
+ || defined (BOXMODEL_CUBEREVO_MINI2) \
+ || defined (BOXMODEL_CUBEREVO_2000HD) \
+ || defined (BOXMODEL_CUBEREVO_3000HD) \
+ || defined (BOXMODEL_IPBOX9900) \
+ || defined (BOXMODEL_IPBOX99)
 			char vol_chr[64] = "";
 			snprintf(vol_chr, sizeof(vol_chr)-1, "Volume: %d%%", (int)vol);
 			ShowText(vol_chr);
@@ -820,15 +1007,18 @@
 		int pp = (int) round((double) vol * (double) 8 / (double) 100);
 		if(pp > 8) pp = 8;
 
-		if(force_update || oldpp != pp) {
-printf("CVFD::showVolume: %d, bar %d\n", (int) vol, pp);
+		if(force_update || oldpp != pp)
+		{
+			printf("CVFD::showVolume: %d, bar %d\n", (int) vol, pp);
 			int i;
 			int j = 0x00000200;
-			for(i = 0; i < pp; i++) {
+			for (i = 0; i < pp; i++)
+			{
 				ShowIcon((fp_icon) j, true);
 				j /= 2;
 			}
-			for(;i < 8; i++) {
+			for (;i < 8; i++)
+			{
 				ShowIcon((fp_icon) j, false);
 				j /= 2;
 			}
@@ -866,7 +1056,7 @@
 		if(pp > 8) pp = 8;
 
 		if(pp != ppold) {
-//printf("CVFD::showPercentOver: %d, bar %d\n", (int) perc, pp);
+//			printf("CVFD::showPercentOver: %d, bar %d\n", (int) perc, pp);
 			int i;
 			int j = 0x00000200;
 			for(i = 0; i < pp; i++) {
@@ -898,7 +1088,7 @@
 	if(fd < 0) return;
 	if (mode != MODE_AUDIO)
 		return;
-printf("CVFD::showAudioTrack: %s\n", title.c_str());
+	printf("CVFD::showAudioTrack: %s\n", title.c_str());
 	ShowText(title.c_str());
 	wake_up();
 
@@ -967,6 +1155,10 @@
 
 void CVFD::setMode(const MODES m, const char * const title)
 {
+#if defined (BOXMODEL_ADB_BOX)
+	struct vfd_ioctl_data data;
+
+#endif
 	if(fd < 0) return;
 
 	// Clear colon in display if it is still there
@@ -983,7 +1177,38 @@
 		ShowIcon(FP_ICON_COL2, false);
 	}
 #endif
-
+#if defined (BOXMODEL_OCTAGON1008) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429)
+	else if (mode == MODE_STANDBY)
+	{
+		ShowIcon(ICON_COLON2, true);
+	}
+#elif defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7819)
+	else if (mode == MODE_STANDBY)
+	{
+		ShowIcon(ICON_COLON, true);
+	}
+#elif defined (BOXMODEL_FORTIS_HDBOX)
+	else if (mode == MODE_STANDBY)
+	{
+		ShowIcon(ICON_COLON3, true);
+	}
+#elif defined (BOXMODEL_ADB_BOX)
+	else if (mode == MODE_STANDBY)
+	{
+		memset(&data, 0, sizeof(struct vfd_ioctl_data));
+		data.start = 0x01; // power LED
+		data.data[0] = 0;
+		data.data[1] = 0;
+		data.data[3] = 1;  // red
+		data.data[4] = 0;
+		data.length = 5;
+		write_to_vfd(VFDSETLED, &data);
+	}
+#endif
 	if(strlen(title))
 		ShowText(title);
 	mode = m;
@@ -1005,6 +1230,10 @@
 			showPercentOver(percentOver, false);
 			break;
 #endif
+		}
+		if (servicename.size() < 1)
+		{
+			servicename = "                ";
 		}
 		showServicename(servicename);
 		showclock = true;
@@ -1086,7 +1315,20 @@
 int CVFD::getBrightness()
 {
 	//FIXME for old neutrino.conf
-#if defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#if defined (BOXMODEL_OCTAGON1008) \
+ || defined (BOXMODEL_FORTIS_HDBOX) \
+ || defined (BOXMODEL_ATEVIO7500) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_HS7819) \
+ || defined (BOXMODEL_ADB_BOX) \
+ || defined (BOXMODEL_HL101) \
+ || defined (BOXMODEL_VIP1_V1) \
+ || defined (BOXMODEL_VIP1_V2) \
+ || defined (BOXMODEL_VIP2)
 	if(g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS] > 7)
 		g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS] = 7;
 #else
@@ -1108,7 +1350,20 @@
 int CVFD::getBrightnessStandby()
 {
 	//FIXME for old neutrino.conf
-#if defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#if defined (BOXMODEL_OCTAGON1008) \
+ || defined (BOXMODEL_FORTIS_HDBOX) \
+ || defined (BOXMODEL_ATEVIO7500) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_HS7819) \
+ || defined (BOXMODEL_ADB_BOX) \
+ || defined (BOXMODEL_HL101) \
+ || defined (BOXMODEL_VIP1_V1) \
+ || defined (BOXMODEL_VIP1_V2) \
+ || defined (BOXMODEL_VIP2)
 	if(g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS] > 7)
 		g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS] = 7;
 #else
@@ -1129,7 +1385,20 @@
 int CVFD::getBrightnessDeepStandby()
 {
 	//FIXME for old neutrino.conf
-#if defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#if defined (BOXMODEL_OCTAGON1008) \
+ || defined (BOXMODEL_FORTIS_HDBOX) \
+ || defined (BOXMODEL_ATEVIO7500) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_HS7819) \
+ || defined (BOXMODEL_ADB_BOX) \
+ || defined (BOXMODEL_HL101) \
+ || defined (BOXMODEL_VIP1_V1) \
+ || defined (BOXMODEL_VIP1_V2) \
+ || defined (BOXMODEL_VIP2)
 	if(g_settings.lcd_setting[SNeutrinoSettings::LCD_DEEPSTANDBY_BRIGHTNESS] > 7)
 		g_settings.lcd_setting[SNeutrinoSettings::LCD_DEEPSTANDBY_BRIGHTNESS] = 7;
 #else
@@ -1200,14 +1466,29 @@
 	else
 		text.clear();
 #else
-#if defined (BOXMODEL_HS7810A) || defined (BOXMODEL_HS7119) || defined (BOXMODEL_HS7819) || defined (BOXMODEL_CUBEREVO_250HD) || defined (BOXMODEL_IPBOX55)
+#if defined (BOXMODEL_HS7810A) \
+ || defined (BOXMODEL_HS7119) \
+ || defined (BOXMODEL_HS7819) \
+ || defined (BOXMODEL_CUBEREVO_250HD) \
+ || defined (BOXMODEL_IPBOX55)
 	ShowText("    ");
-#elif defined (BOXMODEL_OCTAGON1008) || defined (BOXMODEL_TF7700)
+#elif defined (BOXMODEL_OCTAGON1008) \
+ || defined (BOXMODEL_HS7420) \
+ || defined (BOXMODEL_HS7429) \
+ || defined (BOXMODEL_TF7700) \
+ || defined (BOXMODEL_HL101) \
+ || defined (BOXMODEL_VIP1_V1) \
+ || defined (BOXMODEL_VIP1_V2) \
+ || defined (BOXMODEL_VIP2)
 	ShowText("        ");
-#elif defined (BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_ATEVIO7500)
+#elif defined (BOXMODEL_FORTIS_HDBOX) \
+ || defined (BOXMODEL_ATEVIO7500)
 	ShowText("            ");
-#elif defined (BOXMODEL_IPBOX9900) || defined (BOXMODEL_IPBOX99)
+#elif defined (BOXMODEL_IPBOX9900) \
+ || defined (BOXMODEL_IPBOX99)
 	ShowText("              ");
+#elif defined (BOXMODEL_ADB_BOX)
+	ShowText("               ");
 #elif !defined (BOXMODEL_HS7110)
 	ShowText("                ");
 #endif
@@ -1217,25 +1500,56 @@
 void CVFD::ShowIcon(fp_icon icon, bool show)
 {
 #if !HAVE_DUCKBOX_HARDWARE
-	if(!has_lcd || fd < 0) return;
-//printf("CVFD::ShowIcon %s %x\n", show ? "show" : "hide", (int) icon);
+	if(!has_lcd || fd < 0)
+		return;
+
+//	printf("CVFD::ShowIcon %s %02d\n", show ? "show" : "hide", (int) icon);
 	int ret = ioctl(fd, show ? IOC_FP_SET_ICON : IOC_FP_CLEAR_ICON, icon);
 	if(ret < 0)
 		perror(show ? "IOC_FP_SET_ICON" : "IOC_FP_CLEAR_ICON");
 #else
-#if defined (BOXMODEL_ATEVIO7500) || defined (BOXMODEL_HS7110) || defined (BOXMODEL_HS7810A) || defined (BOXMODEL_HS7119) || defined (BOXMODEL_HS7819)
+    #if defined (BOXMODEL_HS7110)
 	return;
-#endif
+    #endif
 	if (icon == 0)
 		return;
 
+  #if !defined (BOXMODEL_OCTAGON1008) \
+   && !defined (BOXMODEL_FORTIS_HDBOX) \
+   && !defined (BOXMODEL_ATEVIO7500) \
+   && !defined (BOXMODEL_HS7420) \
+   && !defined (BOXMODEL_HS7810A) \
+   && !defined (BOXMODEL_HS7420) \
+   && !defined (BOXMODEL_HS7119) \
+   && !defined (BOXMODEL_HS7429) \
+   && !defined (BOXMODEL_HS7819) \
+   && !defined (BOXMODEL_ADB_BOX) \
+   && !defined (BOXMODEL_HL101) \
+   && !defined (BOXMODEL_VIP1_V1) \
+   && !defined (BOXMODEL_VIP1_V2) \
+   && !defined (BOXMODEL_VIP2)
 	if (active_icon[icon & 0x0F] == show)
 		return;
 	else
 		active_icon[icon & 0x0F] = show;
-
-	//printf("CVFD::ShowIcon %s %x\n", show ? "show" : "hide", (int) icon);
+  #endif
+//	printf("CVFD::ShowIcon %s #%02d\n", show ? "show" : "hide", (int) icon);
+
 	struct vfd_ioctl_data data;
+
+  #if defined (BOXMODEL_ADB_BOX)
+	if ((adb_led == 1 && icon == FP_ICON_CLOCK) || (adb_led == 0 && icon == FP_ICON_REC))
+	{
+		memset(&data, 0, sizeof(struct vfd_ioctl_data));
+		data.start = 0x02;  // timer/REC LED
+		data.data[0] = 0;
+		data.data[1] = 0;
+		data.data[3] = show;
+		data.data[4] = 0;
+		data.length = 5;
+		write_to_vfd(VFDSETLED, &data);
+	}  // and fall through to
+  #endif
 	memset(&data, 0, sizeof(struct vfd_ioctl_data));
 	data.start = 0x00;
 	data.data[0] = icon;
@@ -1249,17 +1563,12 @@
 #ifdef HAVE_DUCKBOX_HARDWARE
 void CVFD::ClearIcons()
 {
-#if defined (BOXMODEL_ATEVIO7500) || defined (BOXMODEL_HS7110) || defined (BOXMODEL_HS7810A) || defined (BOXMODEL_HS7119) || defined (BOXMODEL_HS7819)
+  #if defined (BOXMODEL_HS7110)
 	return;
-#endif
-	for (int id = 0x10; id < FP_ICON_MAX; id++) {
-#if defined (BOXMODEL_OCTAGON1008)
+  #endif
+//	printf("CVFD::ClearIcons, min=%02d, max=%02d\n", FP_ICON_MIN, FP_ICON_MAX);
+	for (int id = FP_ICON_MIN; id < FP_ICON_MAX; id++) {
 		if (id != FP_ICON_USB && id != FP_ICON_HDD)
-#elif defined(BOXMODEL_FORTIS_HDBOX) || defined (BOXMODEL_TF7700)
-		if (id != FP_ICON_USB)
-#else
-		if (id != 0x10 && id != 0x12)
-#endif
 			ShowIcon((fp_icon)id, false);
 	}
 	return;
@@ -1270,8 +1578,14 @@
 	memset(g_str, 0, sizeof(g_str));
 	memcpy(g_str, str, sizeof(g_str)-1);
 
+	if (! str)
+	{
+		printf("CVFD::ShowText: str is NULL!\n");
+		return;
+	}
 	int i = strlen(str);
-	if (i > 63) {
+	if (i > 63)
+	{
 		g_str[60] = '.';
 		g_str[61] = '.';
 		g_str[62] = '.';
@@ -1280,14 +1595,15 @@
 	}
 	ShowNormalText(g_str, false);
 }
+
 void CVFD::repaintIcons()
 {
 	char * model = g_info.hw_caps->boxname;
-	if(strstr(model, "ufs912") || strstr(model, "ufs913"))
+	if (strstr(model, "ufs912") || strstr(model, "ufs913"))
 	{
 		bool tmp_icon[16] = {false};
 		printf("VFD repaint icons boxmodel: %s\n", model);
-		for (int i = 0x10; i < FP_ICON_MAX; i++)
+		for (int i = FP_ICON_MIN; i < FP_ICON_MAX; i++)
 		{
 			tmp_icon[i & 0x0F] = active_icon[i & 0x0F];
 			active_icon[i & 0x0F] = false;
@@ -1295,7 +1611,7 @@
 		}
 	}
 }
-#else
+#else // non-duckbox
 void CVFD::ShowText(const char * str)
 {
 	if (fd < 0 || !support_text)
@@ -1335,11 +1651,26 @@
 	if (number < 0)
 		return;
 	
+//	printf("CVFD::ShowNumber: [%d]\n", number);
 #ifdef BOXMODEL_CS_HD2
 	int ret = ioctl(fd, IOC_FP_SET_NUMBER, number);
-	if(ret < 0) {
+	if (ret < 0)
+	{
 		support_numbers = false;
 		perror("IOC_FP_SET_NUMBER");
+	}
+#else
+	{
+		char buffer[VFDLENGTH + 1];
+		int n;
+
+		memset(buffer, 0, sizeof(buffer));
+  #if defined (BOXMODEL_HS7119) || defined (BOXMODEL_HS7810A) || defined (BOXMODEL_HS7819)
+		n = sprintf(buffer, "%04d", number);
+  #else
+		n = sprintf(buffer, "%4d", number);
+  #endif
+		CVFD::ShowText(buffer);
 	}
 #endif
 }
#--- neutrino-mp-tangos/src/driver/pictureviewer/pictureviewer.cpp
#+++ neutrino-mp-tangos/src/driver/pictureviewer/pictureviewer.cpp
#@@ -566,8 +566,10 @@
# 
# 		// add neccessary paths to v_path
# 		v_path.clear();
#+#ifdef ENABLE_LCD4LINUX
# 		if (lcd4l_mode)
# 			v_path.push_back(g_settings.lcd4l_logodir);
#+#endif
# 		v_path.push_back(g_settings.logo_hdd_dir);
# 		if (g_settings.logo_hdd_dir != LOGODIR_VAR)
# 			v_path.push_back(LOGODIR_VAR);
#@@ -669,9 +671,11 @@
# 
# 		for (size_t f = 0; f < v_file.size(); f++)
# 		{
#+#ifdef ENABLE_LCD4LINUX
# 			// process g_settings.lcd4l_logodir
# 			if (lcd4l_mode)
# 				v_path.push_back(g_settings.lcd4l_logodir + "/" + v_file[f] + fileType[i]);
#+#endif
# 			// process g_settings.logo_hdd_dir
# 			v_path.push_back(g_settings.logo_hdd_dir + "/" + v_file[f] + fileType[i]);
# 			// process LOGODIR_VAR
--- neutrino-mp-tangos.org/lib/jsoncpp/jsoncpp.cpp
+++ neutrino-mp-tangos/lib/jsoncpp/jsoncpp.cpp
@@ -101,6 +101,8 @@
 #ifndef JSONCPP_NO_LOCALE_SUPPORT
 #include <clocale>
 #endif
+
+#include <stdio.h>
 
 /* This header provides common string manipulation support, such as UTF-8,
  * portable conversion from/to string...
--- neutrino-mp-tangos.org/src/gui/weather.h
+++ neutrino-mp-tangos/src/gui/weather.h
@@ -116,11 +116,11 @@
 		};
 		std::string getCurrentPressure()
 		{
-			return to_string(current.pressure);
+			return to_string((int)current.pressure);
 		};
 		std::string getCurrentWindSpeed()
 		{
-			return to_string(current.windSpeed);
+			return to_string((int)current.windSpeed);
 		};
 		std::string getCurrentWindBearing()
 		{
@@ -158,7 +158,7 @@
 		{
 			if (i > (int)v_forecast.size())
 				i = (int)v_forecast.size();
-			return to_string(v_forecast[i].windSpeed);
+			return to_string((int)v_forecast[i].windSpeed);
 		};
 		std::string getForecastWindBearing(int i = 0)
 		{
--- neutrino-mp-tangos.org/src/gui/movieplayer.cpp
+++ neutrino-mp-tangos/src/gui/movieplayer.cpp
@@ -82,10 +82,10 @@
 #include <iconv.h>
 #include <libdvbsub/dvbsub.h>
 #include <hardware/audio.h>
-#ifdef ENABLE_GRAPHLCD
-#include <driver/nglcd.h>
+//#ifdef ENABLE_GRAPHLCD
+//#include <driver/nglcd.h>
 bool glcd_play = false;
-#endif
+//#endif
 #include <gui/widget/stringinput_ext.h>
 #include <gui/screensetup.h>
 #include <gui/widget/msgbox.h>
@@ -1446,15 +1446,15 @@
 #if HAVE_SH4_HARDWARE
 	old3dmode = frameBuffer->get3DMode();
 #endif
-#ifdef ENABLE_GRAPHLCD
-	nGLCD::MirrorOSD(false);
-	if (p_movie_info)
-		nGLCD::lockChannel(p_movie_info->channelName, p_movie_info->epgTitle);
-	else {
-		glcd_play = true;
-		nGLCD::lockChannel(g_Locale->getText(LOCALE_MOVIEPLAYER_HEAD), file_name.c_str(), file_prozent);
-	}
-#endif
+//#ifdef ENABLE_GRAPHLCD
+//	nGLCD::MirrorOSD(false);
+//	if (p_movie_info)
+//		nGLCD::lockChannel(p_movie_info->channelName, p_movie_info->epgTitle);
+//	else {
+//		glcd_play = true;
+//		nGLCD::lockChannel(g_Locale->getText(LOCALE_MOVIEPLAYER_HEAD), file_name.c_str(), file_prozent);
+//	}
+//#endif
 	pthread_t thrStartHint = 0;
 	if (is_file_player) {
 		showStartingHint = true;
@@ -1672,14 +1672,14 @@
 
 	while (playstate >= CMoviePlayerGui::PLAY)
 	{
-#ifdef ENABLE_GRAPHLCD
-		if (p_movie_info)
-			nGLCD::lockChannel(p_movie_info->channelName, p_movie_info->epgTitle, duration ? (100 * position / duration) : 0);
-		else {
-			glcd_play = true;
-			nGLCD::lockChannel(g_Locale->getText(LOCALE_MOVIEPLAYER_HEAD), file_name.c_str(), file_prozent);
-		}
-#endif
+//#ifdef ENABLE_GRAPHLCD
+//		if (p_movie_info)
+//			nGLCD::lockChannel(p_movie_info->channelName, p_movie_info->epgTitle, duration ? (100 * position / duration) : 0);
+//		else {
+//			glcd_play = true;
+//			nGLCD::lockChannel(g_Locale->getText(LOCALE_MOVIEPLAYER_HEAD), file_name.c_str(), file_prozent);
+//		}
+//#endif
 		if (update_lcd || g_settings.movieplayer_display_playtime) {
 			update_lcd = false;
 			updateLcd(g_settings.movieplayer_display_playtime);
@@ -2254,12 +2254,12 @@
 	CScreenSetup cSS;
 	cSS.showBorder(CZapit::getInstance()->GetCurrentChannelID());
 #endif
-#ifdef ENABLE_GRAPHLCD
-	if (p_movie_info || glcd_play == true) {
-		glcd_play = false;
-		nGLCD::unlockChannel();
-	}
-#endif
+//#ifdef ENABLE_GRAPHLCD
+//	if (p_movie_info || glcd_play == true) {
+//		glcd_play = false;
+//		nGLCD::unlockChannel();
+//	}
+//#endif
 	if (iso_file) {
 		iso_file = false;
 		if (umount2(ISO_MOUNT_POINT, MNT_FORCE))
@@ -2379,10 +2379,10 @@
 		}
 		if (!movie_info.channelName.empty() || !movie_info.epgTitle.empty())
 			p_movie_info = &movie_info;
-#ifdef ENABLE_GRAPHLCD
-		if (p_movie_info)
-			nGLCD::lockChannel(p_movie_info->channelName, p_movie_info->epgTitle);
-#endif
+//#ifdef ENABLE_GRAPHLCD
+//		if (p_movie_info)
+//			nGLCD::lockChannel(p_movie_info->channelName, p_movie_info->epgTitle);
+//#endif
 	}
 
 	if (p_movie_info) {
@@ -2910,10 +2910,10 @@
 		tuxtx_pause_subtitle(true);
 		frameBuffer->paintBackground();
 	}
-#ifdef ENABLE_GRAPHLCD
-	if (enable_glcd_mirroring)
-		nGLCD::MirrorOSD(g_settings.glcd_mirror_osd);
-#endif
+//#ifdef ENABLE_GRAPHLCD
+//	if (enable_glcd_mirroring)
+//		nGLCD::MirrorOSD(g_settings.glcd_mirror_osd);
+//#endif
 #endif
 }
 
@@ -2951,9 +2951,9 @@
 {
 #if HAVE_SH4_HARDWARE || HAVE_ARM_HARDWARE || HAVE_MIPS_HARDWARE
 	printf("[CMoviePlayerGui] %s: %s\n", __FUNCTION__, show ? "Show" : "Not show");
-#ifdef ENABLE_GRAPHLCD
-	nGLCD::MirrorOSD(false);
-#endif
+//#ifdef ENABLE_GRAPHLCD
+//	nGLCD::MirrorOSD(false);
+//#endif
 
 	if(!show)
 		return;

