--- ../../../tools/exteplayer3/include/common.h.org
+++ ../../../tools/exteplayer3/include/common.h
@@ -28,6 +28,17 @@
     OutputHandler_t		*output;
     ManagerHandler_t	*manager;
 } Context_t;
+// obi
+extern char* subtext;
+// obi
+int container_ffmpeg_update_tracks(Context_t *context, char *filename, int initial);
 
-int container_ffmpeg_update_tracks(Context_t *context, char *filename, int initial);
+const char* GetGraphicSubPath();
+int32_t GetGraphicWindowWidth();
+int32_t GetGraphicWindowHeight();
+
+void E2iSendMsg(const char * format, ...);
+void E2iStartMsg(void);
+void E2iEndMsg(void);
+
 #endif

